<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
<html><head>
<link rel="stylesheet" type="text/css" href="main.css" />
<title>Ancora</title>
</head>

<body>
<p align="center"><img src="logo.png" alt="Ancora"></p>

<table class="main">
<tr>
<td class="textbox">
  
<p><b>Welcome to Ancora!</b></p>

<p>This site is devoted to the study of <i>highly conserved noncoding elements</i> (HCNEs)
in metazoan genomes. We identify HCNEs as non-exonic regions of high similarity between genome sequences
from distantly related organisms, such as human and fish, and provide tools for studying the distribution
of HCNEs along chromosomes.
</p>

<p>Major peaks of HCNE density along chromosomes most often coincide with developmental regulatory genes.
Our aim with this site is to aid discovery of developmental regulatory genes,
their regulatory domains and their fundamental regulatory elements.
</p>

<p>The name Ancora is an acronym for Atlas of Noncoding Conserved Regions in Animals.
Ancora is also a latin word that means anchor, reflecting the ability of HCNEs to constrain
genome evolution and anchor genome alignments.
  
<p>Ancora is maintained by
<a href="http://www.cbu.uib.no/research/lenhard_group/">Boris Lenhard's research group</a>
at Bergen Center for Computational Science and
P&auml;r Engstr&ouml;m at the European Bioinformatics Institute.
Questions and comments can be sent to boris.lenhard at bccs.uib.no or
  engstrom at ebi.ac.uk.</p>
</p>

<p>Please select one of the links on the right to learn more.</p>
 
</td>

<td class="textbox">

  
<p>
Browse HCNEs on the genome of:
<ul style="list-style-type: circle; padding-left: 10px; margin-left: 10px;">
<li><a href="/cgi-bin/gbrowse/hg18">Human</a>
<li><a href="/cgi-bin/gbrowse/mm9">Mouse</a>
<li><a href="/cgi-bin/gbrowse/danRer5">Zebrafish</a>
<li><a href="/cgi-bin/gbrowse/tetNig1">Tetraodon</a>
<li><a href="/cgi-bin/gbrowse/dm3"><i>D. melanogaster</i></a>
<li><a href="/cgi-bin/gbrowse/ce4"><i>C. elegans</i></a>
</ul>
</p>

  
<hr>
  
<p><a href="external_browsers.html">Browse HCNEs in external genome browsers</a></p>

<hr>
  
<p><a href="downloads/">Download HCNE data</a></p>

<hr>
  
<p><a href="methods.html">Methods and references</a></p>

</td></tr>

</table>
<hr>
<table class="maincenter">

<tr>

  <td width="15%"></td>

<td>

  <p align="center"><b>News</b></p>
  <p>
  <b>24 November 2008.</b> Viewing Ancora data in the Ensembl genome browser is now a breeze. We have updated the Ancora DAS server so that it is compatible with Ensembl's new web interface that was released last week. <a href="external_browsers.html">Read about how to view Ancora data in Ensembl.</a>
  <p>
  <b>6 June 2008.</b> The site has been updated with new genomes (<i>Tetraodon</i>, <i>C. elegans</i>, and the latest mouse assembly) and new HCNE sets (human vs. opossum and human vs. platypus). We have also recomputed all HCNEs following some small changes to the detection procedure. Please see <a href="methods.html">Methods</a> for an up-to-date description of the procedure.</p>
  <p>
  <b>15 February 2008.</b> The <a href="http://genomebiology.com/2008/9/2/R34" target="_blank">Ancora paper</a> is published in Genome Biology. Please cite it if you use Ancora in your work.</p>
  
</td>

  <td width="15%"></td>

</tr>
</table>
<hr>

</body> </html>
