<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML//EN">
<html><head>
<link rel="stylesheet" type="text/css" href="main.css" />
<title>Ancora - Methods and References</title>
</head>

<body>
<p align="center"><a href="/" align="center"><img src="/logo.png" border=0 alt="Ancora"></a></p>

<h3>Methods and references</h3>

<table class="main">
<tr>
<td class="textbox">
  
<p><b>Publication</b></p>

The main reference for Ancora is:

<p>
Engstr&ouml;m et al. (2008)
Ancora: a web resource for exploring highly conserved noncoding elements and their association with developmental regulatory genes.
<i>Genome Biol.</i> 9:R34.
<a href="http://genomebiology.com/2008/9/2/R34" target="_blank">
Free full text
</a>
</p>
  
Please refer to this paper for a detailed description of Ancora and cite it if you use Ancora in your work.

<p><b>Background</b></p>
  
Metazoan genomes contain highly conserved noncoding elements (HCNEs) clustered in large arrays
that tend to span developmental regulatory genes and define regulatory domains maintained in
evolution.
Many of these HCNEs have been shown to function as developmental enhancers in reporter gene assays.
An efficient way to locate HCNE arrays is to study distributions of HCNE density along chromosomes.
Major peaks of HCNE density most often coincide with fundamental developmental
regulators situated in large blocks of conserved synteny.
For more reading, we recommend the following papers in addition to the
<a href="http://genomebiology.com/2008/9/2/R34" target="_blank">Ancora paper</a>:

<p><u>Reviews</u></p>
  
<ul>

  <li>
  Ahituv et al. (2004)
  Exploiting human-fish genome comparisons for deciphering gene regulation.
  <i>Hum. Mol. Genet.</i> 13:R261-6.
  <a href="http://hmg.oxfordjournals.org/cgi/content/full/13/suppl_2/R261" target="_blank">
  Free full text
  </a>
  
  <li>
  Kleinjan and van Heyningen (2005)
  Long-range control of gene expression: emerging mechanisms and disruption in disease.
  <i>Am. J. Hum. Genet.</i> 76:8-32.
  <a href="http://www.pubmedcentral.nih.gov/articlerender.fcgi?artid=1196435" target="_blank">
  Free full text
  </a>
  
  <li>
  G&oacute;mez-Skarmeta et al. (2006)
  New technologies, new findings and new concepts in the study of vertebrate cis-regulatory sequences.
  <i>Dev. Dyn.</i> 235:870-85.
  <a href="http://www3.interscience.wiley.com/cgi-bin/fulltext/112224571/HTMLSTART" target="_blank">
  Free full text
  </a>

  <li>Becker and Lenhard (2007)
  The random versus fragile breakage models of chromosome evolution: a matter of resolution.
  <i>Mol. Genet. Genomics</i>.
  <a href="http://www.springerlink.com/content/x75113284427553p/" target="_blank">
  Abstract
  </a>
  
</ul>

<p><u>Research papers</u></p>

<ul>

  <li>
  Bejerano et al. (2004)
  Ultraconserved elements in the human genome.
  <i>Science</i> 304:1321-1325.
  <a href="http://www.sciencemag.org/cgi/content/abstract/304/5675/1321" target="_blank">
  Abstract
  </a>
  
  <li>
  Sandelin et al. (2004)
  Arrays of ultraconserved non-coding regions span the loci of key developmental genes in vertebrate genomes.
  <i>BMC Genomics</i> 5:99.
  <a href="http://www.biomedcentral.com/1471-2164/5/99" target="_blank">
  Free full text
  </a>

  <li>
  Woolfe et al. (2005)
  Highly conserved non-coding sequences are associated with vertebrate development.
  <i>PLoS Biol.</i> 3:e7.
  <a href="http://dx.doi.org/10.1371/journal.pbio.0030007" target="_blank">
  Free full text
  </a>

  <li>
  Glazov et al. (2005)
  Ultraconserved elements in insect genomes: a highly conserved intronic sequence implicated in the control of homothorax mRNA splicing.
  <i>Genome Res.</i> 15:800-8.
  <a href="http://www.genome.org/cgi/content/full/15/6/800" target="_blank">
  Free full text
  </a>
  
  <li>
  Pennacchio et al. (2006)
  In vivo enhancer analysis of human conserved non-coding sequences.
  <i>Nature</i> 444:499-502.
  <a href="http://www.nature.com/nature/journal/v444/n7118/abs/nature05295.html" target="_blank">
  Abstract
  </a>

  <li>
  Vavouri et al. (2007)
  Parallel evolution of conserved non-coding elements that target a common set of developmental regulatory genes from worms to humans.
  <i>Genome Biol.</i> 8:R15.
  <a href="http://genomebiology.com/2007/8/2/R15" target="_blank">
  Free full text
  </a>
  
  
  <li>Kikuta et al. (2007)
  Genomic regulatory blocks encompass multiple neighboring genes and maintain conserved synteny in vertebrates.
  <i>Genome Res.</i> 17:545-55.
  <a href="http://www.genome.org/cgi/content/full/17/5/545" target="_blank">
  Free full text
  </a>

  <li>Engstr&ouml;m et al. (2007)
  Genomic regulatory blocks underlie extensive microsynteny conservation in insects.
  <i>Genome Res.</i> 17:1898-908.
  <a href="http://www.genome.org/cgi/content/full/17/12/1898" target="_blank">
  Free full text
  </a>
  
</ul>

<p><b>Software</b></p>
Ancora makes use of
<a href="http://www.mysql.com/">MySQL</a>,
<a href="http://www.apache.org/">Apache</a>,
<a href="http://www.gmod.org/wiki/index.php/Gbrowse">GBrowse</a> and
<a href="http://www.sanger.ac.uk/Software/analysis/proserver/">ProServer</a>.
To show HCNE data in an efficient and flexible manner, we have extended
GBrowse with plugins and custom glyphs.
  
<p><b>HCNE detection</b></p>
<p>
We identify highly conserved elements by scanning pairwise BLASTZ net whole-genome alignments
(nets; <a href="http://www.pnas.org/cgi/content/full/100/20/11484">Kent et al. 2003</a>)
downloaded from the <a href="http://hgdownload.cse.ucsc.edu/downloads.html">UCSC Genome Browser database</a>
for regions with at least <i>I</i> identities over <i>C</i> alignment columns.
We use two different window sizes (<i>C</i>=30 and <i>C</i>=50) for each pair of genomes.
We use identity thresholds (<i>I</i>/<i>C</i>) in the range 70-100% depending on evolutionary distance.
For each pairwise comparison, we scan two sets of nets (one from the perspective of each genome)
in order not to miss elements duplicated in either lineage. We merge highly conserved elements that overlap
on both genomes. We discard elements whose genome coordinates overlap by one or more bp with
<a href="filters.html">annoted exons</a> and <a href="filters.html">known repeats</a>. We then <a href="http://www.genome.org/cgi/content/full/12/4/656">BLAT</a> all remaining elements against the two respective genomes, counting all mapping positions with a sequence identity equal or higher to the identity threshold used in the cross-species alignment scan. We discard any element with more than four mapping locations in a mammalian genome or eight mapping locations in a teleost genome and consider remaining elements HCNEs.
  
</p>
  
<p><b>Detection of synteny blocks between human and zebrafish genomes</b></p>
<p>
We identify human-zebrafish synteny blocks as described in
<a href="http://www.genome.org/cgi/content/abstract/17/5/545">Kikuta et al. 2007</a>.
Briefly, we base the synteny blocks on net alignments (see above) from the zebrafish genome
to the human genome.
Since neutrally evolving sequence typically cannot be aligned between human and zebrafish genomes,
many syntenic regions are divided over several alignments separated by large regions of unaligned sequence.
The net alignment procedure allows gaps to some degree, but to allow for inversions and other local
rearrangements such that syntenic blocks are separated by macrorearrangements rather than smaller
insertions and alignment gaps, we construct a graph based on the highest-scoring (level 1) net
alignments where two alignments (nodes) are connected if they were separated by &lt;100 kb in the
zebrafish genome and &lt;300 kb in the human genome.
We then consider each connected component in the graph to be one synteny block.
We keep the synteny block with most aligned bases to the human genome in cases of block overlap
in the zebrafish genome.
Only blocks with at least 2 kb of aligned sequence are shown in the genome browser. 
</p>
  
<p><b>Detection of synteny blocks among <i>Drosophila</i> genomes</b></p>
<p>
Our method for detecting synteny blocks among flies is described in
<a href="http://www.genome.org/cgi/content/full/17/12/1898">Engstr&ouml;m et al. 2007</a>
and outlined here.

Starting from pairwise chained BLASTZ alignments (chains, downloaded from
the <a href="http://genome.ucsc.edu/">UCSC Genome Browser database</a>) between the <i>D. melanogaster</i>
(<i>Dmel</i>) genome and each of four other <i>Drosophila</i> genomes
(<i>D. annanassase</i>, <i>D. pseudoobscura</i>, <i>D. virilis</i> and <i>D. mojacensis</i>),
we construct pairwise net alignments (nets) by running the program chainNet (part of the
UCSC Genome Browser <a href="http://genome.ucsc.edu/FAQ/FAQlicense">source code</a>)
with option â€“minSpace=1. chainNet filters a set of chains to retain only the best alignment
for each position in one of the genomes
(<a href="http://www.pnas.org/cgi/content/full/100/20/11484">Kent et al. 2003</a>)
The chainNet algorithm tends to prioritize large chains and therefore its output is suitable for
identifying synteny blocks. For each of the four
pairwise genome comparisons, we construct two sets of nets (one from the perspective of each
genome), and use them to filter the chains into a set of reciprocal-best chains (rb-chains) that only
contain alignment columns included in the nets for both genomes.
We construct pairwise synteny blocks from rb-chains in three steps: (1)
Rb-chains are split at gaps that span nets if, within the gap, nets for either genome contain at
least 10 kb in ungapped blocks. We use nets to split rb-chains because they include alignments that
are not reciprocal-best, thus allowing us to capture synteny breaks caused, for example, by species specific
duplications. Only rb-chains that contain at least 10 kb in ungapped blocks after this step are
retained. (2) We classify regions spanned by multiple (nested) rb-chains as being outside synteny
blocks, and truncate nested rb-chains accordingly. Again, we discard rb-chains containing &lt;10 kb in
ungapped blocks. (3) To avoid artificial synteny breaks due to failure to link scaffolds together in
any of the non-<i>Dmel</i> assemblies, we join rb-chains that ae nearest neighbors along the same <i>Dmel</i>
chromosome arm, but on different scaffolds in the non-<i>Dmel</i> assembly, unless the gap between the rb-chains
in either genome contains nets with at least 10 kb of sequence in ungapped blocks (i.e. the
same criterion as used to split chains in step 2 above). The set of rb-chains after this third step
constitutes our pairwise synteny blocks.

<!-- We create 5-way synteny blocks by intersecting the pairwise
synteny blocks based on their coordinates on the <i>Dmel</i> assembly: we assign any two <i>Dmel</i> bases
to the same 5-way synteny block if and only if they are part of the same synteny block in each of the
pairwise comparisons. Finally, we discard 5-way synteny blocks that do not contain at least 10 kb in
ungapped alignments across all pairwise synteny blocks. -->

</p>
  
 
</td>
  
</table>


</body> </html>
