<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN"><html><head>
  <title>ConSIte: Additional Material</title>
  <meta http-equiv="content-type" content="text/html; charset=ISO-8859-1">
</head><body>
<table width="900" border="0">

  <tbody><tr> 
    <td rowspan="3" bgcolor="#ffffff" width="264"><img src="/CONSITE/IMAGES/logga1.PNG" width="252" height="120"> 
    </td>
    <td colspan="3" rowspan="3" bgcolor="#ffffff"> 
      <table width="215" border="1" bgcolor="#ffffff" bordercolor="#999999">
        <tbody><tr> 
          <td> 
            <p align="center"><font face="Arial, Helvetica, sans-serif" size="5"><i><b>Additional 
              material</b></i></font></p>
            <ul>
              <li><font face="Arial, Helvetica, sans-serif"><b><i>Biology behind 
                the ConSite algorithm</i></b></font></li>
              <li><i><b><font face="Arial, Helvetica, sans-serif">Evolutionary 
                filtering and evolutionary distance: which promoter pairs to use</font></b></i></li>
              <li><i><b><font face="Arial, Helvetica, sans-serif">Examples of 
                ConSite analysis on non-mammalien species</font></b></i></li>
              <li><b><i><font face="Arial, Helvetica, sans-serif">Comparison between 
                ConSite and <a href="http://pga.lbl.gov/rvista.html">rVISTA</a></font></i></b></li>
            </ul>
            <p>&nbsp;</p>
          </td>
        </tr>
      </tbody></table>
    </td>
    <td width="113">&nbsp;</td>
    <td width="100">&nbsp;</td>
    <td width="166">&nbsp;</td>
  </tr>
  <tr> 
    <td width="113">&nbsp;</td>
    <td width="100">&nbsp;</td>
    <td width="166">&nbsp;</td>
  </tr>
  <tr> 
    <td width="113" height="223">&nbsp;</td>
    <td width="100" height="223">&nbsp;</td>
    <td width="166" height="223">&nbsp;</td>
  </tr>
  <tr> 
    <td width="264">&nbsp;</td>
    <td width="67">&nbsp;</td>
    <td width="40">&nbsp;</td>
    <td width="120">&nbsp;</td>
    <td width="113">&nbsp;</td>
    <td width="100">&nbsp;</td>
    <td width="166">&nbsp;</td>
  </tr>
  <tr> 
    <td colspan="4" width="550"> 
      <p><font face="Arial, Helvetica, sans-serif"><b><i>Biology behind the ConSite 
        algorithm</i></b></font></p>
      <p><font size="3">Computational detection of transcription factor binding 
        sites in genomic sequence is difficult due to the limited binding specificity 
        of most transcription factors, as evidenced in a high ratio of false to 
        true predictions. </font></p>
      <p><font size="3">One approach of growing popularity is based on the preferential 
        conservation of functional sequences over the course of evolution by selective 
        pressure; termed "phylogenetic footprinting" . Mutations are more likely 
        to be disruptive if they appear in functional sites, resulting in a measurable 
        difference in the rate of evolution between functional and non-functional 
        segments in the genome. The ConSite system reports those putative transcription 
        factor binding sites both situated in conserved regions and located as 
        pairs of sites in equivalent positions in alignments between two orthologous 
        sequences. This approach results in a significant improvement of transcription 
        factor binding sites detection due to an increased signal/noise ratio<b>(Figure 
        1)</b>.</font></p>
      <p><i><b><font face="Arial, Helvetica, sans-serif">Evolutionary filtering 
        and evolutionary distance</font></b></i></p>
      <p><font size="3">It should be noted that evolutionary rates vary dramatically 
        between genes and the choice of species is an important consideration 
        in "phylogenetic footprinting" studies <b>(Figure 1)</b>. Too great an 
        evolutionary distance can result in regulatory alterations or difficulty 
        in aligning short patches of similarity between long sequences. Inadequate 
        evolutionary distance does not significantly improve the overall specificity 
        of predictions</font></p>
    </td>
    <td width="113">&nbsp;</td>
    <td width="100">&nbsp;</td>
    <td width="166">&nbsp;</td>
  </tr>
  <tr> 
    <td colspan="6" bordercolor="#FFFFFF" height="816"> 
      <p>&nbsp;</p>
      <p><img src="/CONSITE/IMAGES/fig1_1.png" width="600" height="722"></p>
    </td>
    <td width="166" height="816">&nbsp;</td>
  </tr>
  <tr> 
    <td colspan="6" bordercolor="#FFFFFF"> 
      <table width="600" border="1" bordercolor="#cccccc">
        <tbody><tr> 
          <td> 
            <p><b><font size="2"><i>Figure 1</i></font></b> </p>
            <p><i><font size="2">Cross-species comparisons of the beta-globin 
              gene promoter: </font></i></p>
            <p><i><font size="2">(A) Analysis of the human promoter without phylogenetic 
              filtering generates numerous predictions, most of which are biologically 
              irrelevant; </font></i></p>
            <p><i><font size="2">(B) Comparison with the chicken promoter fails 
              to detect conserved sites (screened with the artificially low conservation 
              cutoff of 25%);</font></i></p>
            <p><i><font size="2"> (C) Comparison with the mouse promoter sequence 
              identifies conserved sites, including a documented GATA site from 
              Cao and Moi, Ped.Res. 51:415-21, 2002 (boxed); </font></i></p>
            <p><i><font size="2">(D) Comparison with the cow promoter identifies 
              more conserved sites; </font></i></p>
            <p><i><font size="2">(E) Comparison to the Macaque monkey (Macaca 
              cynomolgus) promoter results in a plot similar to the single sequence 
              analysis. Unless indicated, all plots were generated using all available 
              matrices from vertebrates, with 70% conservation cutoff and 85% 
              transcription factor score threshold settings.</font></i></p>
          </td>
        </tr>
      </tbody></table>
      <p>&nbsp;</p>
    </td>
    <td width="166">&nbsp;</td>
  </tr>
  <tr> 
    <td width="264" align="left" valign="top">&nbsp;</td>
    <td colspan="6">&nbsp;</td>
  </tr>
  <tr> 
    <td colspan="4" width="" 550=""> 
      <p>&nbsp;</p>
      <p><i><b><font face="Arial, Helvetica, sans-serif">Examples of ConSite analysis 
        on non-mammalien species</font></b></i></p>
      <p><font size="3">Comparison of genomic sequences from related species is 
        an effective method for the identification of segments likely to mediate 
        a specific biological function. While the work presented in our paper 
        (submitted) focuses on mammalian sequence comparisons, there is no limitation 
        within the ConSite system precluding studies of other organisms. </font></p>
      <p><font size="3">Below, we present sample analysis pairs from nematodes 
        <b>(Figure 2)</b> and insects <b>(Figure3)</b> </font></p>
      <p>&nbsp;</p>
      <p>&nbsp;</p>
    </td>
    <td width="113">&nbsp;</td>
    <td width="100">&nbsp;</td>
    <td width="166">&nbsp;</td>
  </tr>
  <tr> 
    <td width="264"> 
      <p><b><font size="2">Figure 2</font></b></p>
      <p><font size="2"><i>Cuticle protein of Drosophila melanogaster (-1000 to 
        +50) vs the same from D.pseudoobscura run against all insect profiles 
        (no IC limit) from JASPAR (score threshold 85%, conservation cutoff 70%) 
        </i></font></p>
    </td>
    <td colspan="6"><img src="/CONSITE/IMAGES/drosophila.png" width="537" height="534" border="1"></td>
  </tr>
  <tr> 
    <td width="264">&nbsp;</td>
    <td width="67">&nbsp;</td>
    <td width="40">&nbsp;</td>
    <td width="120">&nbsp;</td>
    <td width="113">&nbsp;</td>
    <td width="100">&nbsp;</td>
    <td width="166">&nbsp;</td>
  </tr>
  <tr> 
    <td width="264"> 
      <p><b><font size="2">Figure 3</font></b></p>
      <p><font size="2"><i>Acetylcholinesterase-1 of C.elegans (-1028 to +50) 
        vs the same from C.briggsae), run against all vertebrate and insect profiles 
        with IC&gt;6 from JASPAR (score threshold 80%, conservation cutoff 70%) </i></font></p>
    </td>
    <td colspan="6"><img src="/CONSITE/IMAGES/Ce_Cb_ace-1_tc80_vert_ins.png" width="582" height="500" border="1"></td>
  </tr>
  <tr> 
    <td colspan="4"> 
      <p><font size="6"><b><i><font face="Arial, Helvetica, sans-serif">rVISTA 
        -ConSite comparison</font></i></b></font></p>
      <p>&nbsp;</p>
      <p><font size="5"><b><i><font face="Arial, Helvetica, sans-serif">Test set-up 
        </font></i></b></font></p>
      <p><a href="http://pga.lbl.gov/rvista.html">rVISTA</a> was subject to the 
        same testing procedure and data as ConSite (described in manuscript).</p>
      <p>Worth consideration is that rVISTA is given several advantages during 
        this test:</p>
      <p><i>Circularity</i></p>
      <p>It is not clarified wether the TRANSFAC matrix collection is independent 
        from the sites we search for. Thereby, rVISTA has an advantage due to 
        circularity in the testing procedure.</p>
      <p><i>Redundancy</i></p>
      <p>In addition, rVISTA includes a redundant matrix collection, thus they 
        have multiple models for several of the transcription factors. In our 
        experience with the Transfac database, many of these models are overly 
        specific (and sensitive to circularity problems) as they are built with 
        only a few sites</p>
      <p><i>Evolutionary distance measurements</i></p>
      <p>The available test sets are ill-suited to explore greater evolutionary 
        distances. In greater evolutionary distances, the alignment algorithms 
        probably favor ConSite even further, as VISTA is optimized for rapid alignment 
        of more similar pairs of sequences. ConSite also allows users to submit 
        alignments from any algorithm through a standard format (ClustalW), which 
        makes it more flexible for future improvements in alignment methods.</p>
      <p>&nbsp;</p>
      <p><b><i><font color="#000000" size="4" face="Arial, Helvetica, sans-serif">Sensitivity 
        testing</font></i></b></p>
      <p>In essence, the sensitivity is measured by analyzing how many of the 
        known sites in the set that are found using the appropriate models. Since 
        there is no possibility for users to view matrix models within rVISTA, 
        identification of appropriatee matrix models was based on matrix names 
        and biological knowledge.</p>
      <p><b>At 70% matrix threshold and 70% conservation cutoff, (all other settings 
        default), ConSite locates 62.5% of the true sites, while rVISTA only can 
        find 52.5%.</b></p>
      <p>&nbsp;</p>
      <p><font size="4"><b><i><font face="Arial, Helvetica, sans-serif">Selectivity</font></i></b></font></p>
      <p>We measure how well the orthologous sequence analysis can filter out 
        false positives compared to single sequence analysis. The total number 
        of predictions using respective full matrix collection is used as an approximation 
        of the number of false positives, since most predicted sites are not biologically 
        relevant. There is no functionality for single sequence comparison within 
        the rVISTA system, instead we have aligned two identical sequences and 
        used that as input. Since rVISTA only uses the other sequence for identifying 
        regions of conservation, the results should be identical to single sequence 
        analysis.</p>
      <p>The total number of predicted sites using orthologous analysis divided 
        by the same results using a single sequence is used as a measure of the 
        filtering efficiency. Since this is an internal test, differences in matrix 
        collection size are not an issue. </p>
      <p><b>At the same settings as above, the ratio for ConSite is 13.6% , implying 
        that about 87% of the noise is removed in the anlysis.</b></p>
      <p><b>The corresponding test for rVISTA generates a ratio of 16.6%.</b></p>
      <p>It is also worth pointing out that rVISTA, partly due to the redundancy 
        and low information content models in their matrix collection generates, 
        on average, <b>4-5 as many predicted sites</b> as ConSite using any setting.</p>
      <p><i><b><font face="Arial, Helvetica, sans-serif" size="4">Conclusion</font></b></i></p>
      <p>User friendliness and functionality aside, ConSite still outperforms 
        rVISTA in both sensitivity and selectivity, despite that rVISTA is given 
        several advantages during the test.</p>
    </td>
    <td width="113">&nbsp;</td>
    <td width="100">&nbsp;</td>
    <td width="166">&nbsp;</td>
  </tr>
</tbody>
</table>

<p>&nbsp;</p>

<p>&nbsp;</p>

<p>&nbsp;</p>

<p>&nbsp;</p>

<p>&nbsp;</p>

<p>&nbsp;</p>

<p>&nbsp;</p>

<br>

</body></html>