#Ancora technical documentation
First version by Pär Engström, April 2008. This version by Ge Tan, May 2013. This document was produced by the [GitHub Flavored Markdown](https://help.github.com/articles/github-flavored-markdown "Markdown").

This document describes the implementation the Ancora web resource. For a general description of Ancora and its user interface, see the [Ancora publication](http://genomebiology.com/2008/9/2/R34 "Ancora publication").

**Table of contents**

*    [Installation](#installation)
	* [Software dependencies](#dependencies)
	* [GBrowse2](#GBrowse2)
	* [Gbrowse2 Advanced Installation](#GBrowse2Ad)
	* [UCSC Genome Browser source and utilities](#UCSC)
	* [ProServer DAS server](#DAS)
	* [Software components](#SoftwareComponents)
*    [Configuration](#configuration)
	* [Ancora Web Resource](#ancoraweb)
	* [Apache Configuration](#apache)
	* [Data Files](#data)
	* [The CNE database](#cnedb)
    * [MySQL account for Perl scripts](#mysql)

<h2 id="installation">Installation</h2>
 This documentation focuses on the Ancora installation on Olifant at csc. Olifant is running the CentOS release 5.8 (Final). However, ancora is supposed to run on other Linux/Unix distribution without too much difficulty.
<h3 id="dependencies">Software dependecies</h3>
 The following softwares are essential for the whole implementation.
 
 *  A httpd server (Currently [Apache](http://httpd.apache.org/docs/2.2/ "Apache") 2.2.3 is used on olifant)
 *  [MySQL](http://dev.mysql.com/downloads/mysql/5.0.html "MySQL") client if you have a dedicated MySQL server. if not, MySQL server is alse required. (MySQL 5.0.95 is used on olifant)
 *  [BioPerl](http://www.bioperl.org/wiki/Installing_BioPerl_on_Unix "BioPerl"). Installation is quite straightforward following the instructions. (BioPerl 1.6.1 is used on olifant.)

<h3 id="GBrowse2">GBrowse2</h3>
This implementation of Ancora utilizes the latest version of GBrowse 2.54. GBrowse 2.X is a complete rewrite of GBrowse 1.X version. There are several advantages compared to the GBrowse 1.X. See the [details](http://gmod.org/wiki/GBrowse_2.0_HOWTO#Installation_from_Source_Code "GBrowse").

Since the GBrowse 2.X is perl-based and all the modules are hosted on [CPAN](http://search.cpan.org/~lds/GBrowse-2.54/), the easiest way to install GBrowse 2 is using the standard Perl module building procedure.

For a smooth installation, please install some prerequisites before launching the GBrowse2 installation. Details are mentioned on the [page](http://gmod.org/wiki/GBrowse_2.0_Prerequisites). It is highly recommened to use the common package managers, Debian(DEB) or RedHat Package Manager(RPM).

After installing the prerequisites, run this command in terminal

```sh
sudo perl -MCPAN -e 'install Bio::Graphics::Browser2'
```
If not all the necessary prerequisites are installed, it will be notified during the test step. 
During the installation, there are some prompted configuration questions about the location of files.
In principle, the default paths are fine and you can customize it to fit your needs.
On olifant, we configrued as follows:

* Directory for GBrowse's config and support files? [/etc/gbrowse2] /opt/www/gbrowse2/conf
* Directory for GBrowse's static images & HTML files? [/var/www/html/gbrowse2] /opt/www/gbrowse2/html
* Directory for GBrowse's temporary data [/var/tmp/gbrowse2] /opt/www/gbrowse2/tmp
* Directory for GBrowse's sessions, uploaded tracks and other persistent data [/var/lib/gbrowse2] /opt/www/gbrowse2/lib
* Directory for GBrowse's example databases [/var/lib/gbrowse2/databases] /opt/www/gbrowse2/databases
* Directory for GBrowse's CGI script executables? [/var/www/cgi-bin/gb2] /opt/www/gbrowse2/cgi-bin
* Internet port to run demo web site on (for demo)? [8000]
* Apache loadable module directory (for demo)? [/etc/httpd/modules]
* User account under which Apache daemon runs? [apache]
* Automatically update Apache config files to run GBrowse? [y]
* Automatically update system config files to run gbrowse-slave? [y]

After the installation, the GBrowse2 demos should be accessable now. 
Type ```YourServersIP/gbrowse2``` in the internet browser and 
try the example database *yeast*.

However, because of some bugs in ```Bio::Graphics 2.33```, 
the tracks generated by plugins might complain some errors like 

```sh
Can't locate object method "height" via package "gdTinyFont"
(perhaps you forgot to load "gdTinyFont"?) at /usr/local/share/perl/5.8.8/Bio/Graphics/Panel.pm line 641. at
/usr/local/lib/perl/5.8.8/Bio/Graphics/Browser2/Render.pm line 3678.
```
As long as Lincoln D. Stein does not fix the bug, just downgrade to the ```Bio::Graphics 2.32```. Here is the trick how to do it.

```sh
wget http://backpan.perl.org/authors/id/L/LD/LDS/Bio-Graphics-2.32.tar.gz
tar xzf Bio-Graphics-2.32.tar.gz
perl Build.PL
./Build
./Build test
sudo ./Build install --uninst 1
```
The "--uninst 1" will make sure that the files for the
newer (2.33) version are removed in case they are installed 
in another directory.

**Note**: For some unknown reasons, the tracks are not displayed on Mac OS 10.8.3 with Firefox 20.0, Safari 6.0.4 and Opera 12.15. However, Chrome 26.0.1410.65 works fine. So far, it works on Windows 7 with all major browsers.

<h3 id="GBrowse2Ad">GBrowse2 Advanced Installation</h3>
This section is optional and incomplete. The advance installation should cover the topics such as *FastCGI*, *User Account Database*, *Displaying Next Generation Sequencing Data*, *Configuring the Uploaded Track Database*.

<h4 id="FastCGI">Running GBrowse under FastCGI</h4>
The idea of FastCGI is to make the script as a long-running process at the first time of the script is requested. By eliminating the startup time for later use of script, the responsiveness of a FastCGI is significantly improved.

To use this facility, you will need a version of Apache equipped with FastCGI support, the module [mod_fcgid](http://httpd.apache.org/mod_fcgid/). For a Debian (DEB) based system,

```sh
sudo apt-get install libapache2-mod-fastcgi libfcgi-perl
```
or for Cent OS,

```sh
cd /etc/yum.repos.d/
wget http://centos.karan.org/kbsingh-CentOS-Extras.repo
sudo vim /etc/yum.repos.d/kbsingh-CentOS-Extras.repo 
# set gpgcheck to 0 and enabled to 1 in the [kbs-CentOS-Testing] section.
yum install mod_fcgid
# Down the latest atomic-release rpm from http://www6.atomicorp.com/channels/atomic/centos/5/x86_64/RPMS/
# Install atomic-release rpm:
rpm -Uvh atomic-release*rpm
yum install fcgi-perl
```

The configuration of GBrowse2's FastCGI for Ancora will be addressed later.

**Note**: Because the olifant has a old version of *Apache* and *mod_fcgid*. 
You may need to adapt the GBrowse2 conf file for Apache 
(/etc/httpd/conf.d/gbrowse2.conf on olifant) 
to the older version of *mod_fcgid* with the names mapping 
described on the [page](http://httpd.apache.org/mod_fcgid/mod/mod_fcgid.html).

<h4 id="User">User Account Database</h4>
To be implemented

<h4 id="">Displaying Next Generation Sequencing Data</h4>
To be implemented

<h4 id="">Configuring the Uploaded Track Database</h4>
To be implemented

<h3 id="UCSC">UCSC Genome Browser source and utilities</h3>
The UCSC Genome Browser source code is needed to compile the C program 
for detecting CNEs, which makes use of several library functions 
from the UCSC source. 
The UCSC Genome Browser source also contains a collection of useful utilities, 
some of which are needed at steps below. 
It is recommended to compile all the UCSC utilities by following the instructions in the README file in the UCSC source package. 
Short story on olifant,

```sh
export MACHTYPE=x86_64
export MYSQLINC=/usr/include/mysql
export MYSQLLIBS="/usr/lib64/mysql/libmysqlclient.a -lz"
export PATH=$HOME/bin/$MACHTYPE:$PATH
mkdir -p $HOME/bin/$MACHTYPE.
wegt http://hgdownload.cse.ucsc.edu/admin/jksrc.zip
unzip jksrc.zip
cd kent/src
make libs
cd utils/
make
```

<h3 id="DAS">ProServer DAS server</h3>
To be implemented

<h3 id="SoftwareComponents">Software components</h3>
Ancora requires two software packages developed in Boris Lenhard’s group: 
cne and AT. The cne package contains Perl libraries, 
C source code and Perl and bash scripts for detecting and 
working with CNEs, as well as Perl libraries and CGI scripts 
directly underlying the Ancora web resource. 
The AT package is a collection of Perl modules related to genome 
and transcriptome sequence analysis. 
They were largely implemented by Pär Engström during his PhD studies 
but also include contributions from several other past 
and present members of the Lenhard group. 
Ancora only uses a few well-tested modules from the AT package, 
which also includes many modules in early stages of development.

The cne and AT packages are maintained in the CSC repository. 
To retrieve the most recent versions, do:

```sh
git clone git@github.com:ge11232002/CSC.git
```
One copy of the packages should be placed under /opt/www/cne and /opt/www/AT, respectively, 
where the Ancora genome browser will access them from.
This could be done by creating soft links for the *cne* and *AT* under */opt/www/*.
To run scripts that use the modules, 
you will have to add the installation paths to your PERL5LIB environment variable. 
If you use bash as your shell, 
you can do this by adding the following line to the file ~/.bashrc:
```sh
export PERL5LIB=/opt/www/AT/lib:/opt/www/cne/perl_lib:$PERL5LIB
```
(If you want scripts to use a working copy of the modules in a different location, modify the above line accordingly.)
The cne/tools directory in the cne package contains source code for several C programs. 
Ancora currently only needs one of these programs: ceScan, 
the program we use to detect CNE. 
Instructions for how to compile the C programs are in cne/tools/README.txt.

<h2 id="configuration">Configuration</h2>
This section will describe all the necessary configurations step by step
for setting up the Ancora web resource.
<h3 id="ancoraweb">Ancora Web Resource</h3>
Under the repository, there are two other folders *ancora* and *gbrowse2*,
besides the *cne* and *AT* packages.
The *ancora* directory contains html and cgi files 
related to the Ancora web resource,
while the *gbrowse2* directory contains the *conf* files for GBrowse2.
If you are setting up the Ancora web resource, 
you will need to copy some files to the ancora and gbrowse2 directory on your server. 
On olifant, they are ```/opt/www/ancora``` and ```/opt/www/gbrowse2``` 
(As we installed GBrowse2 to this directory).
The following commands create the links there:
```sh
cd ancora
## the html files
ln -s html /opt/www/ancora/
## the cgi files
ln -s cgi-bin /opt/www/ancora/
## the das server settings
ln -s Bio-Das-ProServer /opt/www/ancora/
## put the gbrowse cgi files to ancora cgi-bin folder
ln -s /opt/www/gbrowse2/cgi-bin/* /opt/www/ancora/cgi-bin
## put the gbrowse conf files to the GBrowse's conf folder 
cd gbrowse2/conf
ln -s *.conf /opt/www/gbrowse2/conf
## put the gbrowse plugins to the GBrowse's plugins folder
ln -s plugins/* /opt/www/gbrowse2/conf/plugins/
```

<h3 id="apache">Apache Configuration</h3>
The Apache httpd server is configured to look for html files under
```/var/www/html``` and CGI scripts in ``` /var/www/cgi-bin```.
However, if we use the technique called "VirtualHost",
the html files can be left under ```/opt/www/ancora```.
The mapping of the domain http://ancora.olifant.cscdom.csc.mrc.ac.uk/ 
to this directory is configured in Apache configuration file
```/etc/httpd/conf```.



