<p align="center"><img src="<TMPL_VAR NAME="IMAGE_DIR">/information_title.PNG" width="163" height="44"> 
</p>
<p align="center"><font face="Arial, Helvetica, sans-serif"><i>Information about <i>Consite</i> and associated programs</i></p></font>
<table width="75%" border="0" cellspacing="1" cellpadding="1" align="center">
  <tr> 
    <td height="46" colspan="2" bgcolor="#FFFFCC"> 
      <p align="center"><font size="4" face="Arial, Helvetica, sans-serif">What is <i>Consite</i></font></p>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2"> 
      <p><i>Consite</i> is a web-based tool for detecting transcription factor binding sites 
        in genomic sequences using phylogenetic footprinting. Two orthologous 
        genomic sequences are aligned, and transcription factor binding sites 
        are only reported for those regions in the alignment which transcend a 
        certain treshold of conservation</p>
      <p><i>Consite</i> also provide means for selecting transcription factor binding 
        profiles to scan sequences with by information content, species, domain 
        class. </p>
      <p>In reporting, <i>Consite</i> provides a wide number of options for displaying 
        the result of the scans</p>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2" bgcolor="#FFFFCC"> 
      <p align="center"><font size="4" face="Arial, Helvetica, sans-serif">Features</font></p>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2"> 
      <p><b><i>Input options</i></b></p>
      <p><i>Consite</i> accepts one or two genomic sequences as input, with an option 
        to include cDNA and/or exon postions.</p>
      <p><b><i>Transcription factor selection options</i></b></p>
      <p>Transcription factor binding profiles can be chosen by </p>
      <ul>
        <li>minimal specificity</li>
        <li>name</li>
        <li>species and species supergroup(s)</li>
        <li>domain class</li>
      </ul>
      <p>A list of selected transcription factors can also be downloaded for later 
        upload and analysis. </p>
      <p><b><i>Consite reports options:</i></b></p>
      <p>Three different views can be chosen:</p>
      <p><b>Graphical view </b>showing </p>
      <ul>
        <li>alignments graphically with transcription factor binding site locations 
          indicated, </li>
        <li>percentage identity plots with coding regions highlighted if a cDNA 
          sequence was supplied.</li>
      </ul>
      <p><b>Alignment view</b> showing</p>
      <ul>
        <li>detailed alignment and transcription factor binding sites within the 
          alignment</li>
        <li>a highlighting of conservwed regions</li>
      </ul>
      <p><b>Table view</b> showing</p>
      <ul>
        <li>all reported transcription factor binding sites in tabular form</li>
        <li>indication of coding/non-coding region</li>
      </ul>
      <p>All views can be modified by user by entering</p>
      <ul>
        <li>range of displayed sequence relative to either input sequence</li>
        <li>conservation cutoff</li>
        <li>window size</li>
        <li>min. information cutoff</li>
      </ul>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2" bgcolor="#FFFFCC"> 
      <div align="center"><font  size="4" face="Arial, Helvetica, sans-serif">Programs used in <i>Consite</i></font></div>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2"> 
      <p><i>Consite</i> uses</p>
      <ul>
        <li><a href="http://kisac.cgr.ki.se/cgi-bin/dpb_page.cgi" target="_blank">DPB</a> 
          for aligning sequences. DPB is constructed by Luis Mendoza.</li>
        <li> [a program] for scanning a sequence with a transcription factor binding 
          profile.[a program] is constructed by Wyeth Wasserman</li>
        <li>Perl modules and graphical programs...[fill in what you fancy].by 
          Boris Lenhard</li>
      </ul>
      <p>&nbsp;</p>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2" bgcolor="#FFFFCC">
     
      <div align="center"><font size="4" face="Arial, Helvetica, sans-serif">Transcription factor binding site profiles 
        used in <i>Consite</i></font></div>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2"> 
      <p>The transcription factor binding site profiles used in <i>Consite</i> are all 
        high-quality, non-redundant experimental binding site studies (mostly 
        SELEX-protocol), from multicellular eukaryotes. All binding profiles have 
        been thouroughly checked and binding sites have been realigned using 
        Gibbs sampler programs.</p>
      <p>The database will be made downloadable in the near future.</p>
    </td>
  </tr>
  <tr> 
    <td height="46" colspan="2" bgcolor="#FFFFCC">
      <div align="right"><a href="http://www.cgr.ki.se/cgr/groups/wasserman/index.html" target="_blank"><i><b>Wasserman 
        group page</b></i></a></div>
    </td>
  </tr>
  </font>
</table>
<p>&nbsp;</p>
<p align="center">&nbsp;</p>
