\documentclass{article}
%\VignetteIndexEntry{TFBSTools}
\usepackage[usenames,dvipsnames]{color}
\usepackage[colorlinks=true, linkcolor=Blue, urlcolor=Blue,
   citecolor=Blue]{hyperref}
\usepackage[round]{natbib}

\newcommand{\Rfunction}[1]{{\texttt{#1}}}
\newcommand{\Robject}[1]{{\texttt{#1}}}
\newcommand{\Rpackage}[1]{{\textit{#1}}}
\newcommand{\Rmethod}[1]{{\texttt{#1}}}
\newcommand{\Rfunarg}[1]{{\texttt{#1}}}
\newcommand{\Rclass}[1]{{\textit{#1}}}
\newcommand{\Rcode}[1]{{\texttt{#1}}}

\newcommand{\software}[1]{\textsf{#1}}
\newcommand{\R}{\software{R}}
\newcommand{\IRanges}{\Rpackage{IRanges}}
\usepackage{amsmath}

\title{The \textbf{CNEr} package overview}
\author{Ge Tan}

\begin{document}
\SweaveOpts{concordance=TRUE}
\maketitle
<<initialize, echo=FALSE>>=
options(width=70)
@

\tableofcontents

\section{Introduction}
Conserved noncoding elements (CNEs) are pervasive class of elements 
clustering around genes with roles in development and differentiation in Metazoa. 
While many have been shown to act as long-range developmental enhancers, 
the source of their extreme conservation remains unexplained. 
To study the evolutionary dynamics of these elements 
and their relationship to the genes around which they cluster, 
it is essential to be able to produce genome-wide sets of these elements 
for a large number of species comparisons, 
each with multiple size and conservation thresholds.

This \Rcode{CNEr} package aims to detect CNEs and visualise them along the genome.
For performance reasons, the implementation of CNEs detection 
and corresponding I/O functions are primarily written as C extenstions to R. 
We have used CNEr to produce sets of CNEs by scanning pairwise whole-genome net alignments with multiple reference species, 
each with two different window sizes and a range of minimum identity thresholds. 
Then, to pinpoint the boundaries of CNE arrays, 
we compute the CNE densities as the percentages of length 
covered by CNEs within a user specified window size. 
Finally, we describe a novel visualidation method using horizon plot tracks 
that shows a superior dynamic range to the standard density plots, 
simultaneously revealing CNE clusters characterized 
by vastly different levels of sequence conservation. 
Such CNE density plots generated using precise locations of CNEs 
can be used to identify genes involved in developmental regulation, 
even for novel genes that are not annotated yet.

\section{Pipeline of the package}
This section will briefly demonstrate the pipeline of CNE identification and visualisation.
More detailed usage of each step will be described in following sections.
\subsection{CNE identification}
\begin{enumerate}
  \item axtNets: The axtNet files can be downloaded from UCSC or generated by itself
  \item scan alignments: The regions with minimal \Rcode{I} identities over \Rcode{C} columns.
  \item remove elements: Elements that overlap with annotated exons/repeats.
  \item merge elements to get CNEs: Elements that overlap on both genomes.
\end{enumerate}

\subsection{CNE visualisation}
\begin{enumerate}
  \item display parameters: Chromosome, start, end, smooth window size.
  \item horizon plot: Visualise CNE densities.
\end{enumerate}

\section{I/O functions}
\Rcode{CNEr} starts from \href{http://genome.ucsc.edu/goldenPath/help/axt.html}{axt}
net files of two genomes pairwise alignments and bed files for filtering.
UCSC already provides a set of precomputed axt files on 
\url{http://hgdownload.soe.ucsc.edu/downloads.html}
for most of popular genomes.
In case the axt files are not available from UCSC, 
you can always generate the axt net files by following the UCSC wiki
\url{http://genomewiki.ucsc.edu/index.php/Whole_genome_alignment_howto}.

\subsection{axt file}
So far, there is no suitable class to store the axt files in Bioconductor.
Hence we created a \Rcode{S4} class \Rcode{axt} to hold the content from axt files.
Basically, it utilises \Rcode{GRanges} from \Rcode{GenomicRanges} package
and \Rcode{DNAStringSet} from \Rcode{Biostrings} package.

To read axt file into R, 
\Rcode{CNEr} provides \Rcode{readAxt} function
for highly efficient reading, 
which heavily depends on Kent's utilities source code \citep{Kent:2002bw}.
The axt file can be gzippped or in plain text file.
The alignments between two genomes can also be in one file or in several files, 
such as chr1.hg19.mm10.net.axt.gz, chr2.hg19.mm10.net.axt.gz, etc.

<<rtl-PFMatrix, eval=FALSE, results = hide>>=
axtFileshg19danRer7 = list.files(path="/Users/gtan/CSC/CNEr/axtNet",
                                 pattern=".*hg19\\.danRer7\\.*", full.names=TRUE)
axtshg19danRer7 = readAxt(axtFileshg19danRer7)
@

\subsection{bed files}
The gene annotation information, including exons coordinates, repeats information,
is provided in a three column bed file.
Here we summarise a table of filtering information we use:

<<rtl-PFMatrix, eval=FALSE, results = hide>>=

@


\section{Conclusion}
The following is the session info that generated this vignette:
<<session-info>>=
  sessionInfo()
@

\newpage
\bibliographystyle{jss}
\bibliography{CNEr}

\end{document}
