<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns:css="http://macVmlSchemaUri" xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta name=Title content="JASPAR Documentation">
<meta name=Keywords content="">
<meta http-equiv=Content-Type content="text/html; charset=utf-8">
<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 2008">
<meta name=Originator content="Microsoft Word 2008">
<link rel=File-List href="help_files/filelist.xml">
<title>JASPAR Documentation</title>
<!--[if gte mso 9]><xml>
 <o:DocumentProperties>
  <o:Author>Albin Sandelin</o:Author>
  <o:Template>Normal</o:Template>
  <o:LastAuthor>Albin Sandelin</o:LastAuthor>
  <o:Revision>105</o:Revision>
  <o:TotalTime>217</o:TotalTime>
  <o:Created>2007-09-09T11:35:00Z</o:Created>
  <o:LastSaved>2009-10-09T13:51:00Z</o:LastSaved>
  <o:Pages>5</o:Pages>
  <o:Words>7240</o:Words>
  <o:Characters>41273</o:Characters>
  <o:Company>The Bioinformatics Center, Copenhagen University</o:Company>
  <o:Lines>343</o:Lines>
  <o:Paragraphs>82</o:Paragraphs>
  <o:CharactersWithSpaces>50686</o:CharactersWithSpaces>
  <o:Version>12.0</o:Version>
 </o:DocumentProperties>
 <o:OfficeDocumentSettings>
  <o:AllowPNG/>
 </o:OfficeDocumentSettings>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:WordDocument>
  <w:Zoom>150</w:Zoom>
  <w:SpellingState>Clean</w:SpellingState>
  <w:GrammarState>Clean</w:GrammarState>
  <w:TrackMoves>false</w:TrackMoves>
  <w:TrackFormatting/>
  <w:ValidateAgainstSchemas/>
  <w:SaveIfXMLInvalid>false</w:SaveIfXMLInvalid>
  <w:IgnoreMixedContent>false</w:IgnoreMixedContent>
  <w:AlwaysShowPlaceholderText>false</w:AlwaysShowPlaceholderText>
  <w:Compatibility>
   <w:SplitPgBreakAndParaMark/>
   <w:DontVertAlignCellWithSp/>
   <w:DontBreakConstrainedForcedTables/>
   <w:DontVertAlignInTxbx/>
   <w:Word11KerningPairs/>
   <w:CachedColBalance/>
   <w:UseFELayout/>
  </w:Compatibility>
 </w:WordDocument>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <w:LatentStyles DefLockedState="false" LatentStyleCount="276">
 </w:LatentStyles>
</xml><![endif]-->
<style>
<!--h1
	{mso-bidi-font-size:10pt;
	mso-bidi-font-weight:normal;}
h2
	{mso-bidi-font-size:10pt;
	mso-bidi-font-weight:normal;}
h3
	{mso-bidi-font-size:10pt;
	mso-bidi-font-weight:normal;}
h6
	{mso-bidi-font-size:10pt;
	mso-bidi-font-weight:normal;}
span.MSOCOMMENTREFERENCE
	{mso-bidi-font-size:9pt;}
span.HTMLPREFORMATTEDCHAR
	{mso-bidi-font-size:10pt;
	mso-style-locked:yes;}
span.COMMENTTEXTCHAR
	{mso-style-locked:yes;}
span.COMMENTSUBJECTCHAR
	{mso-bidi-font-size:10pt;
	mso-style-locked:yes;}
span.HEADING1CHAR
	{mso-bidi-font-size:16pt;
	mso-style-locked:yes;}
span.HEADING2CHAR
	{mso-bidi-font-size:13pt;
	mso-style-locked:yes;}
span.HEADING3CHAR
	{mso-bidi-font-size:10pt;
	mso-style-locked:yes;}
span.HEADING6CHAR
	{mso-bidi-font-size:10pt;
	mso-style-locked:yes;}
span.HEADING1CHAR
	{mso-style-locked:yes;}
span.HEADING2CHAR
	{mso-style-locked:yes;}
span.HEADING3CHAR
	{mso-style-locked:yes;}
span.HEADING6CHAR
	{mso-style-locked:yes;}
span.HTMLPREFORMATTEDCHAR
	{mso-style-locked:yes;}
span.COMMENTTEXTCHAR
	{mso-style-locked:yes;}
span.COMMENTSUBJECTCHAR
	{mso-style-locked:yes;}
span.TITLECHAR
	{mso-bidi-font-size:10pt;
	mso-style-locked:yes;}
span.HEADING1CHAR
	{mso-style-locked:yes;}
span.HEADING2CHAR
	{mso-style-locked:yes;}
span.HEADING3CHAR
	{mso-style-locked:yes;}
span.HEADING6CHAR
	{mso-style-locked:yes;}
span.HTMLPREFORMATTEDCHAR
	{mso-style-locked:yes;}
span.COMMENTTEXTCHAR
	{mso-style-locked:yes;}
span.TITLECHAR
	{mso-style-locked:yes;}
span.COMMENTSUBJECTCHAR
	{mso-style-locked:yes;}
span.HEADING1CHAR
	{mso-style-locked:yes;}
span.HEADING2CHAR
	{mso-style-locked:yes;}
span.HEADING3CHAR
	{mso-style-locked:yes;}
span.HEADING6CHAR
	{mso-style-locked:yes;}
span.HTMLPREFORMATTEDCHAR
	{mso-style-locked:yes;}
span.COMMENTTEXTCHAR
	{mso-style-locked:yes;}
span.TITLECHAR
	{mso-style-locked:yes;}
span.COMMENTSUBJECTCHAR
	{mso-style-locked:yes;}

 /* Font Definitions */
@font-face
	{font-family:"Courier New";
	panose-1:2 7 3 9 2 2 5 2 4 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Times;
	panose-1:2 0 5 0 0 0 0 0 0 0;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Calibri;
	panose-1:2 15 5 2 2 2 4 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
@font-face
	{font-family:Cambria;
	panose-1:2 4 5 3 5 4 6 3 2 4;
	mso-font-charset:0;
	mso-generic-font-family:auto;
	mso-font-pitch:variable;
	mso-font-signature:3 0 0 0 1 0;}
 /* Style Definitions */
p.MsoNormal, li.MsoNormal, div.MsoNormal
	{mso-style-parent:"";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
h1
	{mso-style-link:"Heading 1 Char";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	mso-outline-level:1;
	font-size:24.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;
	font-weight:bold;}
h2
	{mso-style-link:"Heading 2 Char";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	mso-outline-level:2;
	font-size:18.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;
	font-weight:bold;}
h3
	{mso-style-link:"Heading 3 Char";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	mso-outline-level:3;
	font-size:13.5pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;
	font-weight:bold;}
h6
	{mso-style-link:"Heading 6 Char";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	mso-outline-level:6;
	font-size:7.5pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;
	font-weight:bold;}
p.MsoCommentText, li.MsoCommentText, div.MsoCommentText
	{mso-style-link:"Comment Text Char";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
span.MsoCommentReference
	{mso-ansi-font-size:9.0pt;}
p.MsoTitle, li.MsoTitle, div.MsoTitle
	{mso-style-name:"Title\,title";
	mso-style-link:"Title Char\,title Char";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
a:link, span.MsoHyperlink
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
a:visited, span.MsoHyperlinkFollowed
	{color:blue;
	text-decoration:underline;
	text-underline:single;}
p
	{margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";}
pre
	{mso-style-link:"HTML Preformatted Char";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt;
	font-size:10.0pt;
	font-family:Courier;
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:Courier;}
p.MsoCommentSubject, li.MsoCommentSubject, div.MsoCommentSubject
	{mso-style-parent:"Comment Text";
	mso-style-link:"Comment Subject Char";
	mso-style-next:"Comment Text";
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
span.Heading1Char
	{mso-style-name:"Heading 1 Char";
	mso-style-locked:yes;
	mso-style-link:"Heading 1";
	mso-ansi-font-size:16.0pt;
	font-family:Calibri;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:major-bidi;
	color:#345A8A;
	font-weight:bold;}
span.Heading2Char
	{mso-style-name:"Heading 2 Char";
	mso-style-locked:yes;
	mso-style-link:"Heading 2";
	mso-ansi-font-size:13.0pt;
	font-family:Calibri;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	font-weight:bold;}
span.Heading3Char
	{mso-style-name:"Heading 3 Char";
	mso-style-locked:yes;
	mso-style-link:"Heading 3";
	mso-ansi-font-size:10.0pt;
	font-family:Calibri;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:major-bidi;
	color:#4F81BD;
	font-weight:bold;}
span.Heading6Char
	{mso-style-name:"Heading 6 Char";
	mso-style-locked:yes;
	mso-style-link:"Heading 6";
	mso-ansi-font-size:10.0pt;
	font-family:Calibri;
	mso-ascii-font-family:Calibri;
	mso-ascii-theme-font:major-latin;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:major-fareast;
	mso-hansi-font-family:Calibri;
	mso-hansi-theme-font:major-latin;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:major-bidi;
	color:#244061;
	font-style:italic;}
span.HTMLPreformattedChar
	{mso-style-name:"HTML Preformatted Char";
	mso-style-locked:yes;
	mso-style-link:"HTML Preformatted";
	mso-ansi-font-size:10.0pt;
	font-family:"Courier New";
	mso-ascii-font-family:"Courier New";
	mso-hansi-font-family:"Courier New";}
span.CommentTextChar
	{mso-style-name:"Comment Text Char";
	mso-style-locked:yes;
	mso-style-link:"Comment Text";
	font-family:Times;
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;}
span.TitleChar
	{mso-style-name:"Title Char\,title Char";
	mso-style-locked:yes;
	mso-style-link:"Title\,title";
	mso-ansi-font-size:10.0pt;
	font-family:Times;
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;}
span.CommentSubjectChar
	{mso-style-name:"Comment Subject Char";
	mso-style-locked:yes;
	mso-style-parent:"Comment Text Char";
	mso-style-link:"Comment Subject";
	mso-ansi-font-size:10.0pt;
	font-family:Times;
	mso-ascii-font-family:Times;
	mso-hansi-font-family:Times;
	font-weight:bold;}
p.authors, li.authors, div.authors
	{mso-style-name:authors;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
p.source, li.source, div.source
	{mso-style-name:source;
	margin:0cm;
	margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:10.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Times;
	mso-fareast-font-family:"Times New Roman";
	mso-fareast-theme-font:minor-fareast;
	mso-hansi-font-family:Times;
	mso-bidi-font-family:"Times New Roman";
	mso-bidi-theme-font:minor-bidi;}
span.journalname
	{mso-style-name:journalname;}
span.SpellE
	{mso-style-name:"";
	mso-spl-e:yes;}
span.GramE
	{mso-style-name:"";
	mso-gram-e:yes;}
@page Section1
	{size:612.0pt 792.0pt;
	margin:72.0pt 90.0pt 72.0pt 90.0pt;
	mso-header-margin:35.4pt;
	mso-footer-margin:35.4pt;
	mso-paper-source:0;}
div.Section1
	{page:Section1;}
-->
</style>
<!--[if gte mso 10]>
<style>
 /* Style Definitions */
table.MsoNormalTable
	{mso-style-name:"Table Normal";
	mso-tstyle-rowband-size:0;
	mso-tstyle-colband-size:0;
	mso-style-noshow:yes;
	mso-style-parent:"";
	mso-padding-alt:0cm 5.4pt 0cm 5.4pt;
	mso-para-margin:0cm;
	mso-para-margin-bottom:.0001pt;
	mso-pagination:widow-orphan;
	font-size:12.0pt;
	font-family:"Times New Roman";
	mso-ascii-font-family:Cambria;
	mso-ascii-theme-font:minor-latin;
	mso-hansi-font-family:Cambria;
	mso-hansi-theme-font:minor-latin;}
</style>
<![endif]--><!--[if gte mso 9]><xml>
 <o:shapedefaults v:ext="edit" spidmax="1027"/>
</xml><![endif]--><!--[if gte mso 9]><xml>
 <o:shapelayout v:ext="edit">
  <o:idmap v:ext="edit" data="1"/>
 </o:shapelayout></xml><![endif]-->
</head>

<body bgcolor=white lang=EN-US link=blue vlink=blue style='tab-interval:36.0pt'>

<div class=Section1>

<h1><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>JASPAR DOCUMENTATION<o:p></o:p></span></h1>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR is a collection of transcription factor DNA-binding
preferences, modeled as matrices. These can be converted into Position Weight
Matrices (<span class=SpellE>PWMs</span> or <span class=SpellE>PSSMs</span>),
used for scanning genomic sequences. <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR is the only database with this scope where the data can be
used with no restrictions (open-source). For a comprehensive review of models
and how they can be used, please see the following reviews <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=10812473&amp;query_hl=1"><b
style='mso-bidi-font-weight:normal'>DNA binding sites: representation and
discovery</b> Bioinformatics. 2000 Jan<span class=GramE>;16</span>(1):16-23 </a><o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=15131651&amp;query_hl=4"><b
style='mso-bidi-font-weight:normal'>Applied bioinformatics for the
identification of regulatory elements</b> Nat Rev Genet. 2004 Apr<span
class=GramE>;5</span>(4):276-87</a> <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-size:12.0pt;font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<h1 style='text-align:justify;text-justify:inter-ideograph'><a
name=mozTocId2533></a><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>1 JASPAR DATABASES<o:p></o:p></span></h1>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>JASPAR is a collection of smaller databases,
which have different goals. Most scientists use JASPAR CORE. <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<h2><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.1 Version control<o:p></o:p></span></h2>

<p class=MsoNormal><span class=GramE><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>Since<span
style="mso-spacerun: yes">&nbsp; </span>the</span></span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> 2009 release, all matrix models have version numbers. This is
primarily to keep track of improvements – which can be anything from correcting
typos to actually making a new model based on new data. Version control works
as follows:<span style="mso-spacerun: yes">&nbsp; </span>IDs are based on a
stable ID, and a version number, so that the whole ID is<span
style="mso-spacerun: yes">&nbsp; </span>[stable ID]<span class=GramE>.[</span>version].
The stable ID follows a certain transcription factor, or other logic unit such
as a dimmer pair. For instance, the stable ID for the factor GATA1 is MA0035.
However, the GATA1 matrix has been updated with new data, so there are two
versions: GATA1.1 and GATA1.2. Per default, only the latest version is shown,
but it is possible to list all versions of a matrix with the same stable ID. <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<h2><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.2 The JASPAR CORE database<o:p></o:p></span></h2>

<p style='text-align:justify;text-justify:inter-ideograph'><a
name=mozTocId908308></a><a name="JASPAR_CORE"></a><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>The
JASPAR CORE database contains a <span class=SpellE>curated</span>,
non-redundant set of profiles from published articles. All profiles are derived
from published collections of experimentally defined transcription factor
binding sites for multi-cellular eukaryotes. The database represents a <span
class=SpellE>curated</span> collection of target sequences. The binding sites
were historically determined either in SELEX experiments, or by the collection
of data from the experimentally determined binding regions of actual regulatory
regions; this distinction is clearly marked in the profiles' annotation. A
number of new high-throughput techniques, like <span class=SpellE>ChIP-seq</span>,
can also be used.<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>One of the central goals with JASPAR_CORE is to give the <span
class=GramE>single,</span> “best” model for each transcription factor. This
means that the database is non-redundant in the sense that there are not many
models for the same factor (with some few exceptions motivated by biological
complexity, such as <span class=SpellE>dimers</span> and/or splice forms)<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The prime <span class=GramE>difference to similar resources
(TRANSFAC, etc) consist</span> of the open data access, non-redundancy and
quality: JASPAR CORE is a smaller set that is non-redundant and <span
class=SpellE>curated</span>.<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR_CORE is what most scientists mean when referring to JASPAR
in manuscripts. <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>For convenience, JASPAR_CORE is divided by larger groups of
species. This distinction is only used in the web interface and, optionally, in
the download section. <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>What annotation data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> <o:p></o:p></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. CORE matrices always have a <span
  class=SpellE>MAnnnn</span> IDs. Version<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of the transcription factor. As
  far as possible, the name is based on the standardized <span class=SpellE>Entrez</span>
  gene symbols. In the case the model describes a transcription factor hetero-<span
  class=SpellE>dimer</span>, two names are concatenated, such as RXR<span
  class=GramE>::</span>VDR. In a few cases, different splice forms of the same
  gene have different binding specificity: in this case the splice form
  information is added to the name, based on the relevant literature.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Class<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Family<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural sub-class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Species<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The species source for the sequences, in
  Latin. Linked to the NCBI Taxonomic browser. The actual database entries are
  the NCBI tax IDs – the <span class=SpellE>latin</span> conversion is only in
  the web interface. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Tax_group</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Group of species, currently consisting of 4
  larger groups: vertebrate, insect, plant, chordate<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Acc<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>A representative protein accession number
  in <span class=SpellE>Genbank</span> for the transcription factor.&nbsp;Human
  takes precedence if several <span class=GramE>exists</span>. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Type<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Methodology used for matrix <span
  class=GramE>construction&nbsp; </span><o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> link to the relevant publication reporting the sites used in
  the mode building<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:9'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Pazar_tf_id</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>A link to the PAZAR database<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:10;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Comment<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>For some matrices, a curator&nbsp;comment
  is added<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>When seeking models for specific factors or structural classes, or
if experimental evidence is paramount<b style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<h2><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.3 JASPAR COLLECTIONS<o:p></o:p></span></h2>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR collections are collections of matrices that are useful but
do not fit under the JASPAR CORE scope. Examples include splice forms,
computationally derived patterns with no liked transcription factors<span
class=GramE>,<span style="mso-spacerun: yes">&nbsp; </span>meta</span>-models
etc. <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<h3><a name=mozTocId57243></a><a name="JASPAR_FAM"></a><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>1.3.1.&nbsp;&nbsp;&nbsp;
JASPAR FAM <o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The JASPAR FAM database consists of 11 models describing shared
binding properties of structural classes of transcription factors. These types
of models can be called &quot;familial profiles&quot;, &quot;consensus
matrices&quot; or <span class=SpellE>metamodels</span>. The models have two
prime benefits: <span class=GramE>1)Since</span> many factors have similar
target sequences, we often experience multiple predictions at the same
locations that correspond to the same site. <span class=GramE>This type of
models reduce</span> the complexity of the results. <span class=GramE>2)The</span>
models can be used to classify newly derived profiles (or project what type of
structural class its cognate transcription factor belongs to). The construction
of the models is based on the JASPAR CORE database and described in detail in <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>Sandelin A, Wasserman WW. <i style='mso-bidi-font-style:normal'>Constrained
binding site diversity within families of transcription factors enhances
pattern discovery bioinformatics</i> <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=15066426&amp;query_hl=11"><b
style='mso-bidi-font-weight:normal'>J Mol Biol. 2004 Apr 23<span class=GramE>;338</span>(2):207-15</b></a>.
A recent, comprehensive study of familial binding profiles and associated
methods is available in <i style='mso-bidi-font-style:normal'>(that <span
class=SpellE>plos</span> paper by <span class=SpellE>maohney</span> et al</i>)<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<o:p></o:p></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>A&nbsp;unique identifier for each model.
  FAM matrices always <span class=GramE>have&nbsp; <span class=SpellE>MFnnnn</span></span>
  IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of model. In this database, models
  were built by first partitioning JASPAR CORE matrices <span class=GramE>into&nbsp;
  structural</span> classes – therefore, the names are essentially structure
  class names<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The source article (always <a
  href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=15066426&amp;query_hl=11"><b
  style='mso-bidi-font-weight:normal'>J Mol Biol. 2004 Apr 23<span class=GramE>;338</span>(2):207-15</b></a>)<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Included models<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The JASPAR CORE matrices used to construct
  the model<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Type<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Always “<span class=SpellE>Metamodel</span>”<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>When searching large genomic sequences with no prior knowledge. <span
class=GramE>For classification of new user-supplied profiles.</span><b
style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<h3><a name=mozTocId957953></a><a name="JASPAR_PHYLOFACTS"></a><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>1.3.2. &nbsp;&nbsp;&nbsp; JASPAR PHYLOFACTS<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The JASPAR PHYLOFACTS database consists of 174 profiles that were
extracted from <span class=SpellE>phylogenetically</span> conserved gene
upstream elements. <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>For a detailed description, see <span class=SpellE>Xie</span> et
al., <i style='mso-bidi-font-style:normal'>Systematic discovery of regulatory
motifs in human promoters and 3' <span class=SpellE>UTRs</span> by comparison
of several mammals<span class=GramE>.<span style='font-style:normal'>,</span></span></i><a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=15735639&amp;query_hl=3">
Nature 434, 338-345 (2005)</a> and supplementary material <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>In short, the authors used the following strategy. Promoters
(defined as the 4-kb region around the TSS) of human genes from the <span
class=SpellE>RefSeq</span> database were aligned against the genomes of mouse,
rat and dog. Every consensus sequence of length between 6 and 26, defined over
an alphabet of 4 unique (A<span class=GramE>,C,G,T</span>) and 7 degenerate (R,
Y, K, M, S, W, N) nucleotides, was scanned over the alignments. A motif is
regarded as conserved when it appears in the alignment both for the human and
for the other three mammalian species. The conservation rate p is defined as
the number of times a motif is conserved divided by the number of times it
occurs in man only. This conservation rate is compared to the expected
conservation rate p0, estimated from random motifs, which gives the motif
conservation <span class=GramE>score</span> MCS. Only motifs with an MCS&gt;6
were retained, resulting in a list of 174 highly conserved motifs (see
supplementary Table S2 of <span class=SpellE>Xie</span> et al.). The count
matrices for these 174 motifs were extracted from the downloaded alignments.
They were further annotated according to their resemblance with TRANSFAC and
JASPAR CORE motifs. For TRANSFAC, the annotation of <span class=SpellE>Xie</span>
et al. was used. For comparing to the JASPAR CORE matrices, the Pearson
Correlation Coefficient (PCC) was used to define matrix similarity. All
PHYLOFACTS matrices were scanned against the JASPAR CORE matrices, and matrices
were regarded as being similar when PCC&gt;0.8. When multiple hits were found,
only the one with the highest PCC was retained. . <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<b style='mso-bidi-font-weight:normal'>What data does each
entry hold?</b>&nbsp;<o:p></o:p></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. PHYLOFACTS matrices always <span
  class=GramE>have&nbsp; <span class=SpellE>MFnnnn</span></span> IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of model. In this database, models
  are based on over-represented <span class=GramE>words which</span> are
  unique. The name is simply the consensus sequence.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Jaspar</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The JASPAR_CORE motif that has the best
  similarity score when compared to this model. Only hits with a similarity
  score over 0.8 are considered.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Transfac</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The <span class=SpellE>transfac</span>
  (public version) motif that has the best similarity score when compared to
  this model. Only hits with a similarity score over 0.8 are considered.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Sysgroup</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Group of species. Always “mammals”<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Type<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Always “<span class=SpellE>phylogenetic</span>”<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The source article (always <a
  href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=15735639&amp;query_hl=3">&nbsp;Nature
  434, 338-345 (2005)</a>)<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The JASPAR PHYLOFACTS matrices are a mix of motifs corresponding
to motifs for known and undefined transcription factors. They are useful when
one expects that other factors might determine promoter characteristics, such
as structural aspects and tissue specificity. They are highly complementary to
the JASPAR CORE matrices, so are best used in combination with this matrix set.
<o:p></o:p></span></p>

<h3><a name=mozTocId791511></a><a name="JASPAR_POLII"></a><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></h3>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.3.3. &nbsp;&nbsp; JASPAR POLII<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The deluge of novel data presented recently pertaining
transcription start sites motivates computational studies of core promoters.
The JASPAR_POLII sub-database holds known 13 DNA patterns linked to RNA
polymerase II core promoters, such as the <span class=SpellE>Inr</span> and BRE
elements, each based on experimental evidence: each model must be constructed
using 5 or more experimentally verified sites. An important difference to the
transcription factor profiles in JASPAR CORE is that patters here do not
necessarily have a specified protein <span class=SpellE>interactor</span> (See
(15) for a review on core promoter patterns).&nbsp; When possible, profiles
were extended by two nucleotides more than the core motif. We consistently
report positions relative to the TSS as the position of 5’ and 3’ edge of the
matrix.&nbsp;<b style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<o:p></o:p></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. POLII matrices always <span
  class=GramE>have&nbsp; <span class=SpellE>POLnnn</span></span> IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The reported name of the pattern (not
  necessarily the binding protein, if this is known)<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Species<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The species source for the sequences, in
  Latin. “-“ <span class=GramE>generally</span> signifies that several species
  were used in the model construction<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>A link to the relevant publication
  reporting the sites used in the mode building<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Start relative to TSS<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Reported bias (if any) on position relative
  to the dominant transcription start site in the promoter. This is counted
  from the 5' end of the pattern (the left side). As we have added some
  flanking nucleotides, this sometimes is not the exact numbers shown in the
  source publications.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>End relative to TSS<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>See above. Distance is counted from the 3'
  end of the matrix (the right side).<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>When analyzing properties of core promoters.<b style='mso-bidi-font-weight:
normal'> <o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.3.4. JASPAR CNE<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>Highly conserved non-coding elements are a distinctive feature of
metazoan genomes. Many of them can be shown to act as long-range enhancers that
drive expression of genes that are themselves regulators of core aspects of
metazoan development and differentiation. Since they act as regulatory inputs,
attempts at deciphering the regulatory content of these elements have started.
JASPAR CNE is a collection of 233 matrix <span class=GramE>profiles&nbsp;
derived</span> by <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Citation&amp;list_uids=17442748"><span
class=SpellE>Xie</span> et al</a> based on clustering of overrepresented motifs
from human conserved non-coding elements. While the biochemical and biological
role of most of these patterns is still unknown, <span class=SpellE>Xie</span>
et al. have shown that the most abundant ones correspond to known DNA-binding
proteins, most notably insulator-binding protein CTCF. These matrix profiles
will be useful for further characterization of regulatory inputs in long-range
developmental gene regulation in vertebrates.<b style='mso-bidi-font-weight:
normal'><o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>What data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<o:p></o:p></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. NCRNA matrices always <span
  class=GramE>have&nbsp; <span class=SpellE>CNnnnn</span></span> IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of model.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Consensus sequence<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>the</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> consensus sequence of the motif - important as it is the basis
  for clustering over-represented sites in this study<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The source article (always&nbsp;<a
  href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Citation&amp;list_uids=17442748"><span
  class=SpellE>Xie</span> et al</a>)<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>When analyzing properties of potential enhancers.<o:p></o:p></span></p>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h3>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.3.4. JASPAR SPLICE<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>This small collection contains matrix profiles of human canonical
and non-canonical splice sites, as matching <span class=SpellE>donor<span
class=GramE>:acceptor</span></span> pairs. It currently contains only 6 highly
reliable profiles obtained from human genome made by <a
href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Citation&amp;list_uids=15475254">Chong
et al</a><span class=GramE>&nbsp;.</span> In the future, we shall include additional
eukaryotic species, as well as new models for <span class=SpellE>exonic</span>
splicing enhancers (ESE) and inhibitors (ESI).<b style='mso-bidi-font-weight:
normal'><o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>What data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<o:p></o:p></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. SPLICE matrices always <span
  class=GramE>have&nbsp; <span class=SpellE>SPnnnn</span></span> IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of model.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The source article (<span class=GramE>always
  <u>&nbsp;</u></span><a
  href="http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=PubMed&amp;dopt=Citation&amp;list_uids=15475254">Chong
  et al</a>&nbsp;)<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When analyzing splice sites and alternative
splicing <o:p></o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.3.5 JASPAR PBM<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>All the PBM collections are built by <span class=GramE>using<span
style="mso-spacerun: yes">&nbsp; </span>new</span> in-vitro techniques, based
on k-<span class=SpellE>mer</span> microarrays. PBM matrix models have their
own <span class=GramE>database which</span> is specialized for the data: <a
href="http://the_brain.bwh.harvard.edu/uniprobe/"><span class=SpellE>UniPROBE</span>.</a><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>The PBM, collection is the set derived by <a
href="http://www.ncbi.nlm.nih.gov/pubmed/19443739?ordinalpos=2&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_DefaultReportPanel.Pubmed_RVDocSum"><span
class=SpellE>Badis</span> et al</a> from binding preferences of 104 mouse transcription
factors. One profile (IRC900814) was excluded because the transcription factor
could not be identified.<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>What data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<b style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. SPLICE matrices always <span
  class=GramE>have&nbsp; <span class=SpellE>PHnnnn</span></span> IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of model.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Class<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Family<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural sub-class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Species<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The species source for the sequences, in
  Latin. Linked to the NCBI Taxonomic browser. The actual database entries are
  the NCBI tax IDs – the <span class=SpellE>latin</span> conversion is only in
  the web interface. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Tax_group</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Group of species, currently consisting of 4
  larger groups: vertebrate, insect, plant, chordate<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>A link to the relevant publication reporting
  the sites used in the mode building<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Type<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Methodology used for matrix <span
  class=GramE>construction&nbsp; </span><o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Comment<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>For some matrices, a curator&nbsp;comment
  is added<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Where it is important that each matrix was
derived using the same protocol<o:p></o:p></span></p>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h3>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.3.6 JASPAR PBM HOMEO<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>All the PBM collections are built by <span class=GramE>using<span
style="mso-spacerun: yes">&nbsp; </span>new</span> in-vitro techniques, based
on k-<span class=SpellE>mer</span> microarrays. PBM matrix models have their
own <span class=GramE>database which</span> is specialized for the data: <a
href="http://the_brain.bwh.harvard.edu/uniprobe/"><span class=SpellE>UniPROBE</span>.</a><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>The PBM, collection is the set derived by <a
href="http://www.ncbi.nlm.nih.gov/pubmed/18585359?ordinalpos=1&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_DefaultReportPanel.Pubmed_RVDocSum">Berger
et al</a> including 176 profiles from mouse <span class=SpellE>homeodomains</span><b
style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>What data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<o:p></o:p></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. SPLICE matrices always <span
  class=GramE>have&nbsp; <span class=SpellE>PHnnnn</span></span> IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of model.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Class<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Family<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural sub-class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Species<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The species source for the sequences, in
  Latin. Linked to the NCBI Taxonomic browser. The actual database entries are
  the NCBI tax IDs – the <span class=SpellE>latin</span> conversion is only in
  the web interface. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Tax_group</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Group of species, currently consisting of 4
  larger groups: vertebrate, insect, plant, chordate<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>A link to the relevant publication
  reporting the sites used in the mode building<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Type<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Methodology used for matrix <span
  class=GramE>construction&nbsp; </span><o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Comment<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>For some matrices, a curator&nbsp;comment
  is added<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>When should it be used?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Where it is important that each matrix was
derived using the same protocol, focused on <span class=SpellE>homeobox</span>
factors<o:p></o:p></span></p>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h3>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h3>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h3>

<h3><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>1.3.7 JASPAR PBM HLH<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>All the PBM collections are built by <span class=GramE>using<span
style="mso-spacerun: yes">&nbsp; </span>new</span> in-vitro techniques, based
on k-<span class=SpellE>mer</span> microarrays. PBM matrix models have their
own <span class=GramE>database which</span> is specialized for the data: <a
href="http://the_brain.bwh.harvard.edu/uniprobe/"><span class=SpellE>UniPROBE</span>.</a><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>The PBM HLH, collection is the set derived by
<a
href="http://www.ncbi.nlm.nih.gov/entrez/utils/fref.fcgi?PrId=3048&amp;itool=AbstractPlus-def&amp;uid=19632181&amp;nlmid=0413066&amp;db=pubmed&amp;url=http://linkinghub.elsevier.com/retrieve/pii/S0092-8674(09)00519-4">Grove
et al</a>.<span style="mso-spacerun: yes">&nbsp; </span>It holds 19 </span><i
style='mso-bidi-font-style:normal'><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin;mso-fareast-language:KO'>C. <span
class=SpellE>elegans</span></span></i><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin;mso-fareast-language:
KO'> <span class=SpellE>bHLH</span> transcription factor models</span><span
lang=SV style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin;mso-ansi-language:SV'><o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>What data does each entry hold?</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>&nbsp;<b style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<table class=MsoNormalTable border=1 cellpadding=0 width="100%"
 style='width:100.0%;mso-cellspacing:2.0pt;mso-padding-alt:2.0pt 2.0pt 2.0pt 2.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>ID<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>a</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'> unique identifier for each model. SPLICE matrices always <span
  class=GramE>have&nbsp; <span class=SpellE>PHnnnn</span></span> IDs<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:1'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Name<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The name of model.<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:2'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Class<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:3'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Family<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Structural sub-class of the transcription
  factor, based on the <span class=SpellE>TFCaT</span> system<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:4'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Species<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>The species source for the sequences, in
  Latin. Linked to the NCBI Taxonomic browser. The actual database entries are
  the NCBI tax IDs – the <span class=SpellE>latin</span> conversion is only in
  the web interface. <o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:5'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span class=SpellE><span style='font-family:Calibri;mso-ascii-theme-font:
  major-latin;mso-hansi-theme-font:major-latin'>Tax_group</span></span><span
  style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
  major-latin'><o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Group of species, currently consisting of 4
  larger groups: vertebrate, insect, plant, chordate<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:6'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Medline<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>A link to the relevant publication
  reporting the sites used in the mode building<o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:7'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Type<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Methodology used for matrix <span
  class=GramE>construction&nbsp; </span><o:p></o:p></span></p>
  </td>
 </tr>
 <tr style='mso-yfti-irow:8;mso-yfti-lastrow:yes'>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>Comment<o:p></o:p></span></p>
  </td>
  <td style='padding:2.0pt 2.0pt 2.0pt 2.0pt'>
  <p><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
  mso-hansi-theme-font:major-latin'>For some matrices, a curator&nbsp;comment
  is added<o:p></o:p></span></p>
  </td>
 </tr>
</table>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style="mso-spacerun:
yes">&nbsp;</span>When should it be used?</span></b><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Where it is important that each matrix was
derived using the same protocol, focused on <span class=SpellE>bHLH</span>
factors<o:p></o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<h1><a name=mozTocId183674></a><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>2.&nbsp; JASPAR&nbsp;WEB API<o:p></o:p></span></h1>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The JASPAR database can now be reached remotely through a new Web
Service interface. Current functionality includes retrieval of profiles by
name, by identifier and by searching profile annotations. Profiles can be
retrieved as position frequency matrices, position weight matrices or
information content matrices. The purpose of providing an external application
programming interface (API) is to simplify the utilization of JASPAR in
distributed applications and in scientific workflows created in workflow
editors like <span class=SpellE><span class=GramE>Triana</span></span><span
class=GramE>&nbsp;,</span> <a href="http://en.wikipedia.org/wiki/BPEL">BPEL</a>&nbsp;,
or <span class=SpellE>Taverna</span>&nbsp;. Other benefits include platform-
and language independent access, as well as constant up-to-date access to the
database over time. The API is implemented as a WS-I compliant Web service,
identical to the technology used for the services made available through the <a
href="www.embracegrid.info">EMBRACE Network</a> of Excellence, and the Web
service technology chosen by the European Bioinformatics Institute (EBI<span
class=GramE>)&nbsp;.&nbsp;</span> The WSDL describing this service can be found
<a href="http://api.bioinfo.no/wsdl/JasparDB.wsdl">here</a>. Further
information about the Web service is available in the WSDL file, including
example clients in Java&nbsp;and Python.<o:p></o:p></span></p>

<h1><a name=mozTocId139035></a><a name="PERL_API"></a><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h1>

<h1><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>3.JASPAR PERL API (TFBS<span class=GramE>::</span>DB::JASPAR5.pm)<o:p></o:p></span></h1>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR has a fully fledged API in the Perl programming language: <span
class=SpellE><span class=GramE>TFBS.pm</span></span><span class=GramE> ,</span>
which interacts with the <span class=SpellE>mysql</span> version of JASPAR. The
API can be used for a large number of tasks&nbsp; - including searching
sequences and alignments. See the<a href="http://forkhead.cgb.ki.se/TFBS/">
TFBS</a> page and the TFBS<span class=GramE>::</span>DB::JASPAR5 section in the
<a href="http://tfbs.genereg.net/DOC/">manual</a>. JASPAR5.pm is very similar
to JASPAR4 in terms of methods, but is built on another data model ‘under the
hood’ (see below). <b style='mso-bidi-font-weight:normal'><o:p></o:p></b></span></p>

<h3 style='text-align:justify;text-justify:inter-ideograph'><a
name=mozTocId3526></a><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h3>

<h1><a name=mozTocId686927></a><a name=DOWNLOAD></a><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>4.DOWNLOADING
DATABASES<o:p></o:p></span></h1>

<h2><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h2>

<h2><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>4.1 Annotation and matrices<o:p></o:p></span></h2>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR is downloadable in two different formats<span class=GramE>,&nbsp;
from</span> the&nbsp;<a href="../DOWNLOAD">DOWNLOAD</a> link in the start page:
<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span class=SpellE><span
class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>i</span></span></span><span class=GramE><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>)</span><span style='font-size:7.0pt;font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>&nbsp;&nbsp;</span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>flat</span></span><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'> files resulting from the <span
class=SpellE>TFBS::DB::MatrixDir</span> function in the <span class=SpellE>perl</span>
API, which also are easily <span class=SpellE>parsable</span> <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>ii)</span><span style='font-size:7.0pt;font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>&nbsp;</span><span
class=GramE><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>flat</span></span><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>
files corresponding top the <span class=SpellE>mysql</span> tables used
internally in the database, which also are easily <span class=SpellE>parseable</span>
. The create table statements with a brief <span class=SpellE>descrption</span>
are described <span class=GramE>in<span style="mso-spacerun: yes">&nbsp;
</span>section</span> 4.3.<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>iii) <span class=GramE>matrices</span> without annotation (only ID
and name in a <span class=SpellE>fasta</span>-like header)<span
style="mso-spacerun: yes">&nbsp; </span>in JASPAR format <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>You can choose to download either<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span class=SpellE><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>i</span></span><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>) All data (all matrices from all
collections, and all versions if multiple versions of matrices exist)<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>ii) JASPAR CORE only, as<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
class=GramE>ii</span> a) non-redundant, meaning that only the latest matrix
versions are included<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
class=GramE>ii</span> b) redundant, which also includes the older versions<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
class=GramE>ii</span> c) all species or broken up by taxonomic group (like
“vertebrates”). This can be combined with or without redundancy<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>iii) <span class=GramE>all</span> matrices from each JASPAR
collection except CORE, per collection<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<h2><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>4.2 Sites and alignments<o:p></o:p></span></h2>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Downloading sites and alignments</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>In <span class=GramE>the&nbsp; </span><a href="../DOWNLOAD">DOWNLOAD</a>
directory, there is a sites subdirectory which holds one file per matrix model:
it contains all sites used for the construction of the matrix , named <span
class=SpellE>matrixID.version.sites</span>. The format is <span class=SpellE>fasta</span>,
where alignments are implicit – the used sub-parts of sequences are in
capitals. Note that in the majority of cases, this is an interpretation – we
use pattern finders to find the most likely alignment, but this might not
always be the most correct. This is the principal reason we make these
collections available – users can make their own models based on the raw files.
It is also possible to reach these files individually from the detailed
description of matrices within the rest of the web interface (the pop-up
window). <b style='mso-bidi-font-weight:normal'><span style="mso-spacerun:
yes">&nbsp;</span><o:p></o:p></b></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-size:18.0pt;mso-bidi-font-size:10.0pt;font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>4.3
JASPAR5 SQL <span class=GramE>layout</span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-size:18.0pt;mso-bidi-font-size:10.0pt;font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>CREATE TABLE <span class=GramE>MATRIX(</span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>ID
INT NOT NULL AUTO_INCREMENT,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>COLLECTION
VARCHAR (16) DEFAULT '',<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>BASE_ID
VARCHAR (<span class=GramE>16)DEFAULT</span> '' NOT NULL ,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>VERSION
TINYINT DEFAULT <span class=GramE>1<span style="mso-spacerun: yes">&nbsp;
</span>NOT</span> NULL ,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>NAME
VARCHAR (255) DEFAULT '' NOT NULL,<span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>PRIMARY
KEY (ID)<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>);<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>CREATE TABLE MATRIX_<span class=GramE>DATA(</span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>ID
INT NOT NULL,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
class=GramE>row</span> VARCHAR(1) NOT NULL, <o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
class=SpellE><span class=GramE>col</span></span> TINYINT(3) UNSIGNED NOT NULL, <o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
class=SpellE><span class=GramE>val</span></span> float(10,3), <o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>PRIMARY
KEY (ID, row, <span class=SpellE>col</span>)<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>);<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>CREATE TABLE MATRIX_<span class=GramE>ANNOTATION(</span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style="mso-spacerun:
yes">&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span style='mso-tab-count:
1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>ID
INT NOT NULL,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>TAG
<span class=GramE>VARCHAR(</span>255)DEFAULT '' NOT NULL,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>VAL
<span class=SpellE><span class=GramE>varchar</span></span><span class=GramE>(</span>255)
DEFAULT '',<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>PRIMARY
KEY (ID, TAG)<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>);<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>CREATE TABLE MATRIX_<span class=GramE>SPECIES(</span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style="mso-spacerun:
yes">&nbsp;&nbsp;&nbsp;&nbsp; </span><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span><span
style="mso-spacerun: yes">&nbsp;&nbsp;</span>ID INT NOT NULL,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>TAX_ID
<span class=GramE>VARCHAR(</span>255)DEFAULT '' NOT NULL<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>);<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style="mso-spacerun:
yes">&nbsp;</span>CREATE TABLE MATRIX_<span class=GramE>PROTEIN(</span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>ID
INT NOT NULL,<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style='mso-tab-count:1'>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; </span>ACC
<span class=GramE>VARCHAR(</span>255)DEFAULT '' NOT NULL<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>);<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><span style="mso-spacerun: yes">&nbsp;&nbsp;
</span><o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>A few notes:<o:p></o:p></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The MATRIX table contains the fundamental matrix information: base
id, version ID, collection (CORE, etc), and an internal id (<span class=GramE>ID)which</span>
is only used for linking to the other tables. Each matrix model has one row in
this table, which links to the other <span class=GramE>tables<span
style="mso-spacerun: yes">&nbsp; </span>in</span> a one-&gt;many relationship<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>MATRIX_DATA holds the count matrices. Linked by internal ID <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span class=SpellE><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>MATRIX_annotation</span></span><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'> holds custom
annotation for a given matrix, in a tag-value format. Linked by internal ID.<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>MATRIX_SPECIES holds species IDs (NCBI Tax IDs) – possibly many
per model. Linked by internal ID<o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span style="mso-spacerun: yes">&nbsp;</span>MATRIX_PROTEIN holds protein
accession number(s) for each matrix. The reason for having many accession
numbers is that some models are describing <span class=SpellE>dimers</span>. <o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph'><b style='mso-bidi-font-weight:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<h1 style='text-align:justify;text-justify:inter-ideograph'><a
name=mozTocId966272></a><a name=HANDBOOK></a><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>5.JASPAR WEB
HANDBOOK<a name="_msoanchor_3"></a><a href="#_msocom_3"></a></span><span
style='mso-bookmark:_msoanchor_3'></span><span class=MsoCommentReference><span
style='font-size:9.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;display:none;mso-hide:all'>&nbsp;</span></span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p></o:p></span></h1>

<h2 style='text-align:justify;text-justify:inter-ideograph'><a
name=mozTocId881852></a><a name="START_PAGE"></a><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>5.1.</span><span
style='font-size:7.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;font-weight:normal'>&nbsp;&nbsp;&nbsp; </span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>START PAGE<o:p></o:p></span></h2>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>The start page enables fast-browse options to
JASPAR CORE, detailed search options for any of the JASPAR collections, and
also the possibility to compare custom patterns to already existing patterns
(matrices). <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>Browse the JASPAR CORE database directly:</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> <span style="mso-spacerun: yes">&nbsp;</span>Clicking on any of
these picture links gives all the matrices in JASPAR CORE within the taxonomic
group of interest. The last option is to browse by structure (see below). <o:p></o:p></span></p>

<table class=MsoNormalTable border=0 cellspacing=0 cellpadding=0
 style='mso-cellspacing:0cm;background:white;mso-padding-alt:1.0pt 1.0pt 1.0pt 1.0pt'>
 <tr style='mso-yfti-irow:0;mso-yfti-firstrow:yes;mso-yfti-lastrow:yes'>
  <td style='padding:1.0pt 1.0pt 1.0pt 1.0pt'>
  <p class=MsoNormal><o:p>&nbsp;</o:p></p>
  </td>
 </tr>
</table>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>BROWSE &nbsp;&nbsp;tab:</span></b><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'> The whole
selected database will be shown&nbsp; (see the below for information about the
browse page), sorted by the selected attribute (default attribute is ID).<o:p></o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>SEARCH <span class=GramE>BY&nbsp; tab</span>:</span></b><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> &nbsp;selection of subsets of profiles using user set criteria,
use the search by fields.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Multiple inputs are acceptable, if submitted
with commas in between ',' (this will in effect be interpreted as an OR
statement)<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>The criteria will be interpreted from top to
bottom by using the <span class=SpellE><span class=GramE>boolean</span></span>
statement at each row:<o:p></o:p></span></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>an</span></span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> AND statement will perform an intersect between two query results<o:p></o:p></span></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>an</span></span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> OR statement will perform a union of two query results<o:p></o:p></span></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>an</span></span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'> NOT statement will filter out results from the first query<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>ALIGN to custom matrix tab: </span></b><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>In
some cases, it is beneficial to assess similarity to input data (as with using
BLAST for sequence data comparison when using <span class=SpellE>Genbank</span>).
The input profile can consist of actual counts or be normalized (each column
sum =1). Log-odds matrices should be avoided. For an example of the input
format, press the “fill in an example matrix”. The A<span class=GramE>[ ]</span>
etc characters are optional, so:<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:"Courier New"'>A [13 <span
class=SpellE>13</span> 3 1 54 1 <span class=SpellE>1</span> <span class=SpellE>1</span>
0 3 2 <span class=GramE>5 ]</span><br>
C [13 39 5 53 0 1 50 1 0 37 0 17 ]<br>
G [17 2 37 0 <span class=SpellE>0</span> 52 3 0 53 8 37 12 ]<br>
T [11 0 9 0 <span class=SpellE>0</span> <span class=SpellE>0</span> <span
class=SpellE>0</span> 52 1 6 15 20 ]<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><br>
<span class=GramE>is</span> equivalent to<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:"Courier New"'>13 <span
class=SpellE>13</span> 3 1 54 1 <span class=SpellE>1</span> <span class=SpellE>1</span>
0 3 2 5<br>
13 39 5 53 0 1 50 1 0 37 0 17<br>
17 2 37 0 <span class=SpellE>0</span> 52 3 0 53 8 37 12 <br>
11 0 9 0 <span class=SpellE>0</span> <span class=SpellE>0</span> <span
class=SpellE>0</span> 52 1 6 15 20 <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:"Courier New"'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>All profiles in the selected database will be
compared to the input profile, using a modified Needleman-<span class=SpellE>Wunsch</span>
algorithm described in<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Sandelin A, <span class=SpellE>Hoglund</span>
A, <span class=SpellE>Lenhard</span> B, Wasserman WW. Integrated analysis of
yeast regulatory sequences for biologically linked clusters of genes. <a
href="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=12827523&amp;ordinalpos=20&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum"><span
class=SpellE><span style='color:windowtext;text-decoration:none;text-underline:
none'>Funct</span></span><span style='color:windowtext;text-decoration:none;
text-underline:none'> <span class=SpellE>Integr</span> Genomics. 2003 Jul<span
class=GramE>;3</span>(3):125-34 </span></a>and sorted by raw comparison score
(for reference, the maximum score is 2*the width of the smallest matrix in the
compared pair). Both the score and fraction of potential maximal score is
reported.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>It is also possible to input an IUPAC
consensus string instead of a matrix.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-size:13.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:black'><o:p>&nbsp;</o:p></span></p>

<h2 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId129550></a><a name=BROWSE></a><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>5.2.</span><span
style='font-size:7.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;font-weight:normal'>&nbsp;&nbsp;&nbsp; </span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>BROWSE PAGE<o:p></o:p></span></h2>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>This page presents a list of selected
profiles. At the top, it is possible to select a subset of the matrices, much
like in the start page described above.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Note that the columns of this page will
differ between databases, but ID and Name attributes will always be shown, as
well as a sequence logo for each pattern. <b style='mso-bidi-font-weight:normal'>For
detailed information regarding any profile model, press the view link – this
will give a pop-up window with detailed matrix information (see below).<o:p></o:p></b></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>At the right-hand side, a number of
functional analyses can be made, with selected profiles (see the selection
field in the left-most column). Currently, its is possible to <span
class=SpellE>i</span>) Cluster selected matrices into familial binding profiles
using the STAMP tool, ii) create permuted matrices using column shuffling o the
selected <span class=GramE>matrices &nbsp;iii</span>) create random matrices
using a Bayesian sampling procedure, &nbsp;iv) perform basic sequence analysis
(scanning an input sequence with matrices). <o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>These features are described in detail below,
in section d<span class=GramE>) :</span> extended functionality</span>.</p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><o:p>&nbsp;</o:p></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><o:p>&nbsp;</o:p></b></p>

<h2 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId387351></a><a name=SUMMARY></a><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>5.3.</span><span
style='font-size:7.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;font-weight:normal'>&nbsp;&nbsp;&nbsp; </span><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>DETAILED MATRIX INFORMATION<o:p></o:p></span></h2>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>These pop-up pages are results of clicking on
the logos of chosen models in the browse page, and show detailed information
about the model: both annotation data (which is different in different
databases – see respective database entry above), and a sequence logo, and
count matrix<o:p></o:p></span></p>

<p class=MsoNormal><a name=mozTocId819536></a><a name=LOGO></a><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></b></p>

<p class=MsoNormal><b style='mso-bidi-font-weight:normal'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>LOGO</span></b><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>:&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>&nbsp;Sequence logos are graphical
representations of the matrix model, based on information content. The
information content of a matrix column ranges from 0 (no base preference) and 2
(only 1 base used). A sequence logo is basically a <span class=SpellE>barplot</span>
showing the total information content in each position, where the bar is
replaced by stacked letters (A<span class=GramE>,C,G,T</span>), which are sized
and sorted&nbsp; relative to their occurrence. See <a
href="http://www-lmmb.ncifcrf.gov/~toms/paper/primer/latex/index.html">Schneider
et al</a><span class=GramE><span style="mso-spacerun: yes">&nbsp; </span>for</span>
a more comprehensive description.<o:p></o:p></span></p>

<p class=MsoNormal><a name=SVG></a><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>The “Make SVG button” gives an
SVG version logo suitable for publication images&nbsp; (SVG is a vector format
that does not have the pixel edges of <span class=GramE>the .<span
class=SpellE>png’s</span></span> used in browsers – it can be read by many
drawing programs and most web-browsers with proper <span class=SpellE>plugins</span>)
<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><a name=mozTocId526678></a><a name=MATRIX></a><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>COUNT
MATRIX:&nbsp;<o:p></o:p></span></b></p>

<p class=MsoNormal><span class=GramE><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>The
underlying model showing the DNA pattern.</span></span><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'> In
most databases, the cell numbers indicate the number of sequences having base x
in column y. These matrices can be used for a number of different analyses,
including site searching, if suitably converted, See <a
href="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=15131651&amp;ordinalpos=17&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum"><span
style='color:windowtext;text-decoration:none;text-underline:none'>Wasserman and
Sandelin</span></a> for a review.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>The reverse complement <span class=GramE>button
make</span> a reverse complement version of the matrix (as DNA is two-stranded,
the two models are functionally equivalent). If the <span class=SpellE>amtrix</span>
if reverse-complemented, the logo will change accordingly.<o:p></o:p></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal><a name=mozTocId438185></a><a name=STATS></a><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>VERSION
INFORMATION<o:p></o:p></span></b></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>For some transcription factors, there are
multiple models – usually this is due to new data becoming available. Clicking
on this link gives a listing of all the models for the factor in question. <o:p></o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId378719></a><span lang=EN-GB style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman";
mso-ansi-language:EN-GB'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span lang=EN-GB style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin;mso-bidi-font-family:
"Times New Roman";mso-ansi-language:EN-GB'>SITES<o:p></o:p></span></b></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
lang=EN-GB style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman";
mso-ansi-language:EN-GB'>The button in this section will show the sites used
for the matrix construction: either as raw text (<span class=SpellE>fasta</span>)
or as formatted html. The HTML page will show all unique sequences and the
count of them in the original experiment (also see section 4.2). <o:p></o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
lang=EN-GB style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman";
mso-ansi-language:EN-GB'><o:p>&nbsp;</o:p></span></p>

<h2 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId578056></a><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>6. EXTENDED FUNCTIONALITY<o:p></o:p></span></h2>

<h3 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId40025></a><a name=SCAN></a><span class=SpellE><span
style='font-size:10.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>i</span></span><span style='font-size:10.0pt;
font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>.</span><span
style='font-size:10.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;font-weight:normal'>&nbsp;&nbsp;&nbsp;&nbsp; </span><span
style='font-size:10.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>BASIC SEQUENCE ANALYSI<u>S</u><a
name="_msoanchor_5"></a><a href="#_msocom_5"></a></span><span style='mso-bookmark:
_msoanchor_5'><span class=MsoCommentReference><span style='font-size:10.0pt;
display:none;mso-hide:all'>&nbsp;</span></span></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin;color:black'>Using a subset of profiles, a submitted sequence can
be analyzed. Sensitivity and specificity will be affected by the relative score
threshold, by default 80% (</span><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>See <a
href="http://www.ncbi.nlm.nih.gov/sites/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=15131651&amp;ordinalpos=17&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_RVDocSum">Wasserman
and Sandelin</a> <span style='color:black'>for a review on scoring of matrices
to sequences<span class=GramE>) .</span> This is the most basic form of
sequence analysis: dedicated systems such as <a
href="http://asp.ii.uib.no:8090/cgi-bin/CONSITE/consite/"><span class=SpellE>ConSite</span></a>
are preferable for anything more than a casual analysis. <o:p></o:p></span></span></p>

<h3 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId667880></a><span style='font-size:10.0pt;font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><o:p>&nbsp;</o:p></span></h3>

<h3 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-size:10.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;font-weight:normal'>&nbsp;</span><a
name=CLUSTER></a><span style='font-size:10.0pt;font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'>ii.</span><span style='font-size:
10.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin;font-weight:normal'>&nbsp;&nbsp;&nbsp;&nbsp; </span><span
style='font-size:10.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>DYNAMIC CLUSTERING OF MATRICES<o:p></o:p></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The CLUSTER button provides the user with a means of investigating
the relationship between the various matrices. <span class=GramE>This
functionality is provided by the&nbsp;
STAMP</a> tool
available as a <span class=SpellE>webservice</span> at&nbsp;<a
href="http://www.benoslab.pitt.edu/stamp/"></a><span class=MsoHyperlink>http://www.benoslab.pitt.edu/stamp/</span></span></span></p>


Hierarchical clustering is performed on a selected set of matrices using the
UPGMA algorithm with a Pearson Correlation Coefficient distance metric. Then
the optimal number of clusters is selected using a log variant of the <span
class=SpellE>Calinski</span> and <span class=SpellE>Harabasz</span> statistic
(See this <span style='mso-field-code:"HYPERLINK \0022http\:\/\/www\.pubmedcentral\.nih\.gov\/articlerender\.fcgi?artid=1848003\0022"'><u><span
style='color:blue'>link</span></u></span> for details)<span class=GramE>.Finally</span>
the clusters are <span class=SpellE>partioned</span> and a familial binding
profile is created for each cluster using an iterative refinement, multiple
alignment method. Further details can be found in the STAMP <span
style='mso-field-code:"HYPERLINK \0022http\:\/\/nar\.oxfordjournals\.org\/cgi\/content\/full\/gkm272v1\0022"'><u><span
style='color:blue'>manuscript</span></u></span>.</span><span style='font-family:
Times;mso-bidi-font-family:"Times New Roman";mso-bidi-theme-font:minor-bidi;
color:windowtext;text-decoration:none;text-underline:none'> </span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<h3 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId898046></a><a name=RANDOMIZER></a><span style='font-size:10.0pt;
font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></h3>

<h3 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-size:10.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>DYNAMIC RANDOM MATRIX GENERATION</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></h3>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>
PERMUTATION<span class=MsoCommentReference><span
style='mso-ansi-font-size:10.0pt;display:none;mso-hide:all'>NG</span></span></span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>This option
simply shuffles the columns in matrices. <span class=GramE>This can either be
done by just shuffling columns within each selected matrix</span>, or by
shuffling columns <span class=SpellE>almong</span> all selected matrices.</span></a><span
class=MsoHyperlink><b style='mso-bidi-font-weight:normal'><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></b></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>2.</span><span
style='color:windowtext;font-weight:normal;text-decoration:none;text-underline:
none'>&nbsp;&nbsp;&nbsp; </span><span style='color:windowtext;text-decoration:
none;text-underline:none'>SAMPLING</span></a></span></b><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>This feature
of the database enables the users to generate random Position Frequency
Matrices (<span class=SpellE>PFMs</span>) from selected profiles. </span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'><br>
We assume that each column in the profile is independent and described by a
mixture of <span class=SpellE>Dirichlet</span> <span class=SpellE>multinomials</span>
in which the letters are drawn from a multinomial and the multinomial
parameters are drawn from a mixture of <span class=SpellE>Dirichlets</span>.
Within this model each column has its own set of multinomial parameters but the
<span class=GramE>higher level</span> parameters -- those of the mixture prior
is assumed to be common to all <span class=SpellE>Jaspar</span> matrices. We
can therefore use a maximum likelihood approach to learn these from the
observed column counts of all <span class=SpellE>Jaspar</span> matrices. The
maximum likelihood approach automatically ensures that matrices receive a
weight relative to the number of counts it contains.</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'><br>
Drawing samples from the prior distribution will generate <span class=SpellE>PWMs</span>
with the same statistical properties as the <span class=SpellE>Jaspar</span>
matrices as a whole. <span class=SpellE>PWMs</span> with statistical properties
like those of the selected profiles can be obtained by drawing from a posterior
distribution which is proportional to the prior times a multinomial likelihood
term with counts taken from one of the columns of the selected profiles.</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'><br>
Each 4-dimensional column is sampled by the following three-step procedure: 1. <span
class=GramE>draw</span> the mixture component according to the distribution of
mixing proportions, 2. <span class=GramE>draw</span> an input column randomly
from the concatenated selected profiles and 3. <span class=GramE>draw</span>
the probability vector over nucleotides from a 4-dimensional <span
class=SpellE>Dirichlet</span> distribution. The parameter vector alpha of the <span
class=SpellE>Dirichlet</span> is equal to the sum of the count (of the drawn
input<span class=GramE>)&nbsp; and</span> the parameters of the <span
class=SpellE>Dirichlet</span> prior (of the drawn component).&nbsp; .</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'><br>
Draws from a <span class=SpellE>Dirichlet</span> can be obtained in the
following way from Gamma distributed samples:</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>(X1<span
class=GramE>,X2,X3,X4</span>) = (Y1/V,Y2/V,Y3/V,Y4/V) ~ Dir(α1,α2,α3,α4)</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'><br>
<span class=GramE>where</span> V = sum(Yi) ~ Gamma(shape = sum(<span
class=SpellE>αi</span>), scale = 1).</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>3. OUTPUT
FORMATS</span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'><br>
For both and random generating of matrices you have the choice between three
different output formats:</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>Raw - Each
matrix is separated by a <span class=SpellE>fasta</span> like header starting
with the &gt; symbol and then a matrix ID. The count for each base (ACGT) is
specified on its own space separated line where each element corresponds to one
column. The order of the lines for the bases is A<span class=GramE>,C,G</span>
and finally T.</span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>13 <span
class=SpellE>13</span> 3 1 54 1 <span class=SpellE>1</span> <span class=SpellE>1</span>
0 3 2 5 </span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>13 39 5 53 0
1 50 1 0 37 0 17</span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>17 2 37 0 <span
class=SpellE>0</span> 52 3 0 53 8 37 12</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>11 0 9 0 <span
class=SpellE>0</span> <span class=SpellE>0</span> <span class=SpellE>0</span>
52 1 6 15 20</span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>JASPAR - This
is similar to the raw format, having an identical header. The lines for each
base however starts with a label for the nucleotide (A<span class=GramE>,C,G</span>
or T) and then the columns follow enclosed in brackets: [].</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>A [13 <span
class=SpellE>13</span> 3 1 54 1 <span class=SpellE>1</span> <span class=SpellE>1</span>
0 3 2 <span class=GramE>5 ]</span></span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>C [13 39 5 53
0 1 50 1 0 37 0 <span class=GramE>17 ]</span></span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>G [17 2 37 0 <span
class=SpellE>0</span> 52 3 0 53 8 37 <span class=GramE>12 ]</span></span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>T [11 0 9 0 <span
class=SpellE>0</span> <span class=SpellE>0</span> <span class=SpellE>0</span>
52 1 6 15 <span class=GramE>20 ]</span></span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>TRANSFAC -
This is a TRANSFAC-like format having a header starting with &quot;DE&quot;
then the matrix ID, the matrix name and the matrix class. The data itself is
transposed as compared to the other formats, meaning that each line correspond
to a column in the matrix. The column lines start with a number denoting the
column index (counting</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
class=GramE><span style='color:windowtext;text-decoration:none;text-underline:
none'>from</span></span><span style='color:windowtext;text-decoration:none;
text-underline:none'> 0). After that follows tab separated counts for each base
in that column in the order: A<span class=GramE>,C,G</span> and T. After the <span
class=GramE>lines with the counts follows</span> a final line containing the
string: &quot;XX&quot;.</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>DE MA0048&nbsp;&nbsp;&nbsp;
NHLH1&nbsp;&nbsp;&nbsp; <span class=SpellE>bHLH</span></span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>00&nbsp;&nbsp;&nbsp;
13&nbsp;&nbsp;&nbsp; 13&nbsp;&nbsp;&nbsp; 17&nbsp;&nbsp;&nbsp;
11&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>01&nbsp;&nbsp;&nbsp;
13&nbsp;&nbsp;&nbsp; 39&nbsp;&nbsp;&nbsp; 2&nbsp;&nbsp;&nbsp;
0&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>02&nbsp;&nbsp;&nbsp;
3&nbsp;&nbsp;&nbsp; 5&nbsp;&nbsp;&nbsp; 37&nbsp;&nbsp;&nbsp;
9&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>03&nbsp;&nbsp;&nbsp;
1&nbsp;&nbsp;&nbsp; 53&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp;
0&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>04&nbsp;&nbsp;&nbsp;
54&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp;
0&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>05&nbsp;&nbsp;&nbsp;
1&nbsp;&nbsp;&nbsp; 1&nbsp;&nbsp;&nbsp; 52&nbsp;&nbsp;&nbsp;
0&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>06&nbsp;&nbsp;&nbsp;
1&nbsp;&nbsp;&nbsp; 50&nbsp;&nbsp;&nbsp; 3&nbsp;&nbsp;&nbsp;
0&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>07&nbsp;&nbsp;&nbsp;
1&nbsp;&nbsp;&nbsp; 1&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp;
52&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>08&nbsp;&nbsp;&nbsp;
0&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp; 53&nbsp;&nbsp;&nbsp;
1&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>09&nbsp;&nbsp;&nbsp;
3&nbsp;&nbsp;&nbsp; 37&nbsp;&nbsp;&nbsp; 8&nbsp;&nbsp;&nbsp;
6&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>10&nbsp;&nbsp;&nbsp;
2&nbsp;&nbsp;&nbsp; 0&nbsp;&nbsp;&nbsp; 37&nbsp;&nbsp;&nbsp;
15&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>11&nbsp;&nbsp;&nbsp;
5&nbsp;&nbsp;&nbsp; 17&nbsp;&nbsp;&nbsp; 12&nbsp;&nbsp;&nbsp;
20&nbsp;&nbsp;&nbsp; </span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>XX</span></a><o:p></o:p></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><o:p>&nbsp;</o:p></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='margin-left:144.0pt;text-align:justify;text-justify:inter-ideograph;
text-indent:-18.0pt;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
class=MsoCommentReference><span style='font-size:9.0pt;font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin;display:none;
mso-hide:all'><span
style='color:windowtext;text-decoration:none;text-underline:none'>&nbsp;</span></a></span></span><span
class=MsoHyperlink><b style='mso-bidi-font-weight:normal'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin;color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></b></span></p>

<h1 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId548277></a><a name=FAQ></a><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>7.FAQ</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></h1>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>a.</span><span
style='font-size:7.0pt;color:windowtext;font-weight:normal;text-decoration:
none;text-underline:none'>&nbsp;&nbsp;&nbsp; </span><span style='color:windowtext;
text-decoration:none;text-underline:none'>How do I cite JASPAR?</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>It depends on
what you have used it for. If you simply want to acknowledge you used the last
version, use </span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></b></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman"'><span
class=SpellE><span style='color:windowtext;text-decoration:none;text-underline:
none'>Bryne</span></span><span style='color:windowtext;text-decoration:none;
text-underline:none'> JC, <span class=SpellE>Valen</span> E, Tang MH, <span
class=SpellE>Marstrand</span> T, <span class=SpellE>Winther</span> O, <span
class=SpellE>da</span> <span class=SpellE>Piedade</span> I, Krogh A, <span
class=SpellE>Lenhard</span> B, Sandelin A.</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman"'><span
style='color:windowtext;text-decoration:none;text-underline:none'><span
style='mso-field-code:"HYPERLINK \0022http\:\/\/www\.ncbi\.nlm\.nih\.gov\/pubmed\/18006571?ordinalpos=4&amp;itool=EntrezSystem2\.PEntrez\.Pubmed\.Pubmed_ResultsPanel\.Pubmed_DefaultReportPanel\.Pubmed_RVDocSum\0022"'><u><span
style='color:blue'>JASPAR, the open access database of transcription <span
class=GramE>factor-binding</span> profiles: new content and tools in the 2008
update.</span></u></span></span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman"'>Nucleic Acids
Res. 2008 Jan<span class=GramE>;36</span>(Database issue):D102-6. </span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'><a
href="http://www.ncbi.nlm.nih.gov/pubmed/18006571?ordinalpos=4&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_DefaultReportPanel.Pubmed_RVDocSum"></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>Otherwise:</span></a></span></b><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The original
JASPAR paper<span class=GramE>: &nbsp;&nbsp;&nbsp; </span><br>
Sandelin A, <span class=SpellE>Alkema</span> W, <span class=SpellE>Engstrom</span>
P, Wasserman WW, <span class=SpellE>Lenhard</span> B.&nbsp;&nbsp;&nbsp;&nbsp;</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR: an
open-access database for eukaryotic transcription factor binding profiles.<br>
<span style='mso-field-code:"HYPERLINK \0022http\:\/\/www\.ncbi\.nlm\.nih\.gov\/sites\/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=14681366&amp;ordinalpos=3&amp;itool=EntrezSystem2\.PEntrez\.Pubmed\.Pubmed_ResultsPanel\.Pubmed_RVDocSum\0022"'><u><span
style='color:blue'>Nucleic Acids Res. 2004 Jan 1<span class=GramE>;32</span>(Database
issue):D91-4.</span></u></span></span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>The first
extension (JASPAR FAM and PHYLOFACTS collections):</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
class=SpellE><span style='color:windowtext;text-decoration:none;text-underline:
none'>Vlieghe</span></span><span style='color:windowtext;text-decoration:none;
text-underline:none'> D, Sandelin A, De <span class=SpellE>Bleser</span> PJ, <span
class=SpellE>Vleminckx</span> K, Wasserman WW, van Roy F, <span class=SpellE>Lenhard</span>
B.</span></a><span class=MsoHyperlink><span style='color:windowtext;text-decoration:
none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><<span
style='color:windowtext;text-decoration:none;text-underline:none'>A new
generation of JASPAR, the open-access repository for transcription factor
binding site profiles.<br>
<span style='mso-field-code:"HYPERLINK \0022http\:\/\/www\.ncbi\.nlm\.nih\.gov\/sites\/entrez?Db=pubmed&amp;Cmd=ShowDetailView&amp;TermToSearch=16381983&amp;ordinalpos=1&amp;itool=EntrezSystem2\.PEntrez\.Pubmed\.Pubmed_ResultsPanel\.Pubmed_RVDocSum\0022"'><u><span
style='color:blue'>Nucleic Acids Res. 2006 Jan 1<span class=GramE>;34</span>(Database
issue):D95-7.&nbsp;</span></u></span></span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
style='color:windowtext;text-decoration:none;text-underline:none'>Second
expansion (POLII, SPLICE, CNE, many changes in the web service including matrix
permutations)</span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman"'><span
class=SpellE><span style='color:windowtext;text-decoration:none;text-underline:
none'>Bryne</span></span><span style='color:windowtext;text-decoration:none;
text-underline:none'> JC, <span class=SpellE>Valen</span> E, Tang MH, <span
class=SpellE>Marstrand</span> T, <span class=SpellE>Winther</span> O, <span
class=SpellE>da</span> <span class=SpellE>Piedade</span> I, Krogh A, <span
class=SpellE>Lenhard</span> B, Sandelin A.</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman"'><span
style='color:windowtext;text-decoration:none;text-underline:none'><span
style='mso-field-code:"HYPERLINK \0022http\:\/\/www\.ncbi\.nlm\.nih\.gov\/pubmed\/18006571?ordinalpos=4&amp;itool=EntrezSystem2\.PEntrez\.Pubmed\.Pubmed_ResultsPanel\.Pubmed_DefaultReportPanel\.Pubmed_RVDocSum\0022"'><u><span
style='color:blue'>JASPAR, the open access database of transcription <span
class=GramE>factor-binding</span> profiles: new content and tools in the 2008
update.</span></u></span></span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman"'>Nucleic Acids
Res. 2008 Jan<span class=GramE>;36</span>(Database issue):D102-6. </span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>Third
expansion <span class=GramE>( Large</span> expansion of the CORE database,
including yeast and worm matrices. Also includes new PBM collections)</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>JASPAR, the
open access database of transcription factor binding profiles: new and improved
content in the 2010 update</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p class=MsoNormal><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;mso-bidi-font-family:"Times New Roman"'>Nucleic Acids
Res. 2010 Jan; under review </span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>b.</span><span
style='font-size:7.0pt;color:windowtext;font-weight:normal;text-decoration:
none;text-underline:none'>&nbsp;&nbsp;&nbsp; </span><span style='color:windowtext;
text-decoration:none;text-underline:none'>Why are certain sequences not
downloadable&nbsp;from JASPAR_CORE?</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></b></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-size:7.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>&nbsp;</span><span
style='font-size:10.0pt;color:windowtext;text-decoration:none;text-underline:
none'>This is due to historical reasons. JASPAR_CORE was originally built in
order to create familial binding profiles for as many structural classes of
transcription factor classes as possible. In some experimental literature, only
matrices and not sequences are available. For this project, we were forced to
include some matrices to gain coverage of certain binding site classes. For
recent additions, it is a requirement to have the sequences available.</span></a></span><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><br>
<b style='mso-bidi-font-weight:normal'>d.</b></span><span style='font-size:
7.0pt;color:windowtext;text-decoration:none;text-underline:none'>&nbsp;&nbsp;&nbsp;
</span><b style='mso-bidi-font-weight:normal'><span style='color:windowtext;
text-decoration:none;text-underline:none'>Why is not my matrix study included
in JASPAR_CORE?</span></b></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>There are two
principal explanations.<br>
<br>
The most likely is that we were not aware of&nbsp;your work: please let us
know!</span></a><span class=MsoHyperlink><span style='color:windowtext;
text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>The other
possible reason is that the publication did not live up to the demands of the
curators. As we have human <span class=SpellE>curation</span> of all JASPAR
CORE matrices, this is to some degree an arbitrary call – we are happy to
discuss it with you.</span></a><span class=MsoHyperlink><span style='color:
windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>e.</span><span
style='font-size:7.0pt;color:windowtext;font-weight:normal;text-decoration:
none;text-underline:none'>&nbsp;&nbsp;&nbsp; </span><span style='color:windowtext;
text-decoration:none;text-underline:none'>Linking web services to CPU-intensive
services within JASPAR</span></a></span></b><span class=MsoHyperlink><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin;color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>We appreciate
that other services wants to link to JASPAR. However, if <span class=GramE>your
are</span> using the CPU-intensive services (matrix comparison, randomization
or clustering), please ask the maintainers (see contact information below)
before you do this – otherwise your server might be rejected without warning.
In that case, we <b style='mso-bidi-font-weight:normal'>strongly</b>
suggest&nbsp;setting up a local JASPAR database, as the database and resources
are freely available.</span></a><span class=MsoHyperlink><span
style='color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><b
style='mso-bidi-font-weight:normal'><span style='font-family:Calibri;
mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>f.</span><span
style='font-size:7.0pt;color:windowtext;font-weight:normal;text-decoration:
none;text-underline:none'>&nbsp;&nbsp;&nbsp;&nbsp; </span><span
style='color:windowtext;text-decoration:none;text-underline:none'>Who is JASPAR
anyhow? </span></a></span></b><span class=MsoHyperlink><span style='font-family:
Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin;
color:windowtext;text-decoration:none;text-underline:none'><o:p></o:p></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'>JASPAR was
originally the name of a master student project algorithm for comparing matrix
profiles, an obscure tribute to an even more obscure dialog from the Black
Adder episode “The Black Seal” between the Seven Most Evil Men in the Kingdom:</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><a
href="http://www.ncbi.nlm.nih.gov/pubmed/18006571?ordinalpos=4&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_DefaultReportPanel.Pubmed_RVDocSum"></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<pre style='margin-left:36.0pt;text-align:justify;text-justify:inter-ideograph'><i
style='mso-bidi-font-style:normal'><span style='font-family:Calibri;mso-ascii-theme-font:
major-latin;mso-hansi-theme-font:major-latin'><span style="mso-spacerun: yes">&nbsp;</span>- …<span
class=GramE>and</span> with all haste, we will meet at Old <span class=SpellE>Jaspar’s</span> tavern</span></a></span></i><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></pre><pre style='margin-left:
36.0pt;text-align:justify;text-justify:inter-ideograph'><i style='mso-bidi-font-style:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><br>
<span style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>- How is old <span
class=SpellE>Jaspar</span> these days?</span></a></span></i><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></pre><pre style='margin-left:
36.0pt;text-align:justify;text-justify:inter-ideograph'><i style='mso-bidi-font-style:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><br>
<span style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>- Dead.</span></a></span></i><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></pre><pre style='margin-left:
36.0pt;text-align:justify;text-justify:inter-ideograph'><i style='mso-bidi-font-style:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><br>
<span style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>- How?</span></a></span></i><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></pre><pre style='margin-left:
36.0pt;text-align:justify;text-justify:inter-ideograph'><i style='mso-bidi-font-style:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><br>
<span style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp;&nbsp; </span>- I killed him.</span></a></span></i><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></pre><pre style='margin-left:
36.0pt;text-align:justify;text-justify:inter-ideograph'><i style='mso-bidi-font-style:
normal'><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><br>
<span style="mso-spacerun: yes">&nbsp;&nbsp;&nbsp; </span><span class=GramE>[Loud cheer].</span></span></a></span></i><span
class=MsoHyperlink><span style='font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin;color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></pre>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><a
href="http://www.ncbi.nlm.nih.gov/pubmed/18006571?ordinalpos=4&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_DefaultReportPanel.Pubmed_RVDocSum"></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<h1 style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><a
name=mozTocId357255></a><a name=CONTACT></a><span style='font-size:12.0pt;
font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:major-latin'>7 Contact</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></h1>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-size:12.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>We appreciate
feedback – criticism as well as suggestions for new content. <span class=GramE>Development
and supervision of the <span class=SpellE>JASPAr</span> project is coordinated
by Albin Sandelin and Boris <span class=SpellE>Lenhard</span></span>. </span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-size:12.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'>Albin
Sandelin <span class=SpellE>albin</span> AT binf.ku.dk</span></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

<p style='text-align:justify;text-justify:inter-ideograph;tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-size:12.0pt;font-family:Calibri;mso-ascii-theme-font:major-latin;
mso-hansi-theme-font:major-latin'><Boris <span
class=SpellE>Lenhard</span> <span class=SpellE>Boris.Lenhard</span> AT <span
class=SpellE>bccs.uib.no</span></span></a><span class=MsoHyperlink><b
style='mso-bidi-font-weight:normal'><span style='color:windowtext;text-decoration:
none;text-underline:none'><o:p></o:p></span></b></span></span></p>

<p class=MsoNormal style='text-align:justify;text-justify:inter-ideograph;
tab-stops:45.8pt 91.6pt 137.4pt 183.2pt 229.0pt 274.8pt 320.6pt 366.4pt 412.2pt 458.0pt 503.8pt 549.6pt 595.4pt 641.2pt 687.0pt 732.8pt'><span
style='font-family:Calibri;mso-ascii-theme-font:major-latin;mso-hansi-theme-font:
major-latin'><a
href="http://www.ncbi.nlm.nih.gov/pubmed/18006571?ordinalpos=4&amp;itool=EntrezSystem2.PEntrez.Pubmed.Pubmed_ResultsPanel.Pubmed_DefaultReportPanel.Pubmed_RVDocSum"></a><span
class=MsoHyperlink><span style='color:windowtext;text-decoration:none;
text-underline:none'><o:p></o:p></span></span></span></p>

</div>

</body>

</html>
